

  
    
  


  




  


  

<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="Hugo 0.30.2">
    <meta name="theme" content="Tranquilpeak 0.3.1-BETA">
    <title>a simple gene finder</title>
    <meta name="author" content="Guangchuang Yu">
    <meta name="keywords" content="">

    <link rel="icon" href="https://guangchuangyu.github.io/cn/favicon.png">
    

    
    <meta name="description" content="ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œhttp://ibw2011.fmmu.edu.cn/schedule.htm
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚

è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    <meta property="og:description" content="ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œhttp://ibw2011.fmmu.edu.cn/schedule.htm
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚

è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    <meta property="og:type" content="blog">
    <meta property="og:title" content="a simple gene finder">
    <meta property="og:url" content="/2011/07/a-simple-gene-finder/">
    <meta property="og:site_name" content="Guangchuang Yu">
    <meta name="twitter:card" content="summary">
    <meta name="twitter:title" content="Guangchuang Yu">
    <meta name="twitter:description" content="ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œhttp://ibw2011.fmmu.edu.cn/schedule.htm
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚

è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    
      <meta name="twitter:creator" content="@guangchuangyu">
    
    

    
    

    
      <meta property="og:image" content="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">
    

    
    
    

    

    
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.css" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" />
    
    
    <link rel="stylesheet" href="https://guangchuangyu.github.io/cn/css/style-u6mk0ojoywresbx8iepslrmmhl4stuhrsxuwhkpwrkrx7mryjcaimasnk4pi.min.css" />
    
    

    
      
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-77901140-1', 'auto');
ga('send', 'pageview');
</script>

    
    
  </head>

  <body>
    <div id="blog">
      <header id="header" data-behavior="4">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="https://guangchuangyu.github.io/cn/">Guangchuang Yu</a>
  </div>
  
    
      <a class="header-right-picture "
         href="https://guangchuangyu.github.io/cn/#about">
    
    
    
      
        <img class="header-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=90" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="4">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <a href="https://guangchuangyu.github.io/cn/#about">
          <img class="sidebar-profile-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Guangchuang Yu</h4>
        
          <h5 class="sidebar-profile-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/categories">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/tags">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/archives">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>

    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://github.com/GuangchuangYu" target="_blank">
    
      <i class="sidebar-button-icon fa fa-lg fa-github"></i>
      
      <span class="sidebar-button-desc">GitHub</span>
    </a>
  </li>

    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/index.xml">
    
      <i class="sidebar-button-icon fa fa-lg fa-rss"></i>
      
      <span class="sidebar-button-desc">RSS</span>
    </a>
  </li>

    </ul>
  </div>
</nav>

      

      <div id="main" data-behavior="4"
        class="
               hasCoverMetaIn
               ">
        <article class="post" itemscope itemType="http://schema.org/BlogPosting">
          
          
            <div class="post-header main-content-wrap text-left">
  
    <h1 class="post-title" itemprop="headline">
      a simple gene finder
    </h1>
  
  <div class="postShorten-meta post-meta">
  
    <time itemprop="datePublished" datetime="2011-07-10T09:04:46&#43;08:00">
      
  July 10, 2011

    </time>
  
  
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/cn/categories/conference">conference</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/cn/categories/r">R</a>
    
  


</div>

</div>
          
          <div class="post-content markdown" itemprop="articleBody">
            <div class="main-content-wrap">
              <p>ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œ<a href="http://ibw2011.fmmu.edu.cn/schedule.htm">http://ibw2011.fmmu.edu.cn/schedule.htm</a>
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚</p>

<p>è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚</p>

<blockquote>
<p>Course Projects:</p>

<p>Project 1: Implementation of a simple gene finder</p>

<p>GOAL</p>

<p>Build a simple codon-usage based gene finder for finding genes in
E.coli.</p>

<p>Procedure</p>

<p>Collect 100 gene sequences from the bacterium E. coli in the genbank
(<a href="http://www.ncbi.nlm.nihh.gov">http://www.ncbi.nlm.nihh.gov</a>). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:</p>

<p>Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile</p>

<p>Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.</p>

<p>The output should be printed to the standard output as (xxx stands for
the likelihood)</p>

<p>ORF1: xxx ORF2: xxx</p>
</blockquote>

<p></p>

<pre><code class="language-r">
###########################################################
##                                                       ##
##  Dragonstar Course Project (IBW 2011)                 ##
##                                                       ##
##  Project 1: Implementation of a simple gene finder    ##
##  Author Guangchuang your                              ##
##  Xi'an (2011-07-08)                                   ##
##                                                       ##
###########################################################



##' .. content for \description{} (ECgnfinder: An R program to predict prokaryotic genes (e.g. E.coli) using 1st Markov Chain) ..
##'
##' .. content for \details{} ..
##' @title ECgnfinder
##' @param traningSequence : A FASTA file containing multiple sequences for training set
##' @param testingSequence : A FASTA file containing multiple sequences for testing
##' @param backgroundSequence : A FASTA file containing multiple sequences for building the background model
##' @return log likelihood ratio
##' @author Guangchuang Yu
ECgnfinder &lt;- function(trainingSequence, testingSequence, backgroundSequence, print=FALSE) {
    print (&quot;reading training sequence file...&quot;)
    trainingFasta &lt;- readFASTA(trainingSequence)
    trainSeq &lt;- sapply(trainingFasta, function(i) i$seq)

    print(&quot;building Transition Probability Matrix...&quot;)
    TPM &lt;- getTPM(trainSeq)

    print(&quot;building codon usage table...&quot;)
    codonUsage &lt;- getCodonUsage(trainSeq)

    print(&quot;building NucleoTide table...&quot;)
    NtUsage &lt;- getNtUsage(backgroundSequence)


    print(&quot;reading testing sequence file...&quot;)
    testing &lt;- readFASTA(testingSequence)
    testingSeq &lt;- sapply(testing, function(i) i$seq)

    x &lt;- lapply(testingSeq, logLikelihood, transition=TPM, codonusage=codonUsage, background=NtUsage)
    names(x) &lt;- paste(&quot;Sequence&quot;, seq_along(testingSeq), sep=&quot;&quot;)
    if (print) {
        x
    }
    return(x)
}


##' .. content for \description{} (read FASTA file) ..
##'
##' .. content for \details{} ..
##' @title readFASTA
##' @param file: A FASTA file
##' @return a list of list which containing desc and seq.
##' @author Guangchuang Yu
readFASTA &lt;- function(file) {
    x=scan(file, what=&quot;&quot;, sep=&quot;\n&quot;)
    idx &lt;- grep(&quot;&gt;&quot;, x)
    fasta &lt;- list()
    for (i in 1:length(idx)) {
            desc &lt;- x[idx[i]]
            if (i == length(idx)) {
                stop &lt;- length(x)
            } else {
                stop &lt;- idx[i+1]-1
            }
            seq.idx &lt;- seq(idx[i]+1, stop)
            s &lt;- x[seq.idx]
            s &lt;- paste(s, collapse=&quot;&quot;)
            ss &lt;- list(desc=desc, seq=s)
            fasta[i] &lt;- list(ss)
    }
    return(fasta)
    }


##' Build a transition probability matrix from the given Sequence set.
##'
##' @title getTPM
##' @param sequences : A vector of Sequences.
##' @return transition probability matrix
##' @author Guangchuang Yu
getTPM &lt;- function(sequences) {
    codonLength &lt;- 3
    codons &lt;- getCodon()
    
    ## calculate a transition count matrix
    transition &lt;- matrix(1, nrow=length(codons), ncol=length(codons))  ## pseudo count
    colnames(transition) &lt;- codons
    rownames(transition) &lt;- codons

    for (x in 1:length(sequences)) {
        s &lt;- sequences[x]
        len &lt;- nchar(s)
        ## drop the last alphabets which is too short (

        idx &lt;- seq(1, len, codonLength)
        for (i in seq_along(idx)) {
            ## codonI--codonJ
            codonI &lt;- substr(s,idx[i], idx[i]+codonLength-1)
            if (i != length(idx)) {
                codonJ &lt;- substr(s,idx[i+1], idx[i+1]+codonLength-1)
            }
            if (codonI %in% codons &amp; codonJ %in% codons) {
                transition[codonI, codonJ] &lt;- transition[codonI, codonJ] +1
            } else {
                warning(&quot;codon &quot;, codonI, &quot; or &quot;, codonJ, &quot; which does not defined was detected...&quot;)
            }
        }
    }

    ## Calculate Transition Probability Matrix..
    rowsum &lt;- rowSums(transition)
    TPM &lt;- transition/rowsum
    return(TPM)
}

getCodon &lt;- function() {
    codons &lt;- c(&quot;TTT&quot;, &quot;TTC&quot;, &quot;TTA&quot;, &quot;TTG&quot;,
                &quot;CTT&quot;, &quot;CTC&quot;, &quot;CTA&quot;, &quot;CTG&quot;,
                &quot;ATT&quot;, &quot;ATC&quot;, &quot;ATA&quot;, &quot;ATG&quot;,
                &quot;GTT&quot;, &quot;GTC&quot;, &quot;GTA&quot;, &quot;GTG&quot;,
                &quot;TCT&quot;, &quot;TCC&quot;, &quot;TCA&quot;, &quot;TCG&quot;,
                &quot;CCT&quot;, &quot;CCC&quot;, &quot;CCA&quot;, &quot;CCG&quot;,
                &quot;ACT&quot;, &quot;ACC&quot;, &quot;ACA&quot;, &quot;ACG&quot;,
                &quot;GCT&quot;, &quot;GCC&quot;, &quot;GCA&quot;, &quot;GCG&quot;,
                &quot;TAT&quot;, &quot;TAC&quot;, &quot;TAA&quot;, &quot;TAG&quot;,
                &quot;CAT&quot;, &quot;CAC&quot;, &quot;CAA&quot;, &quot;CAG&quot;,
                &quot;AAT&quot;, &quot;AAC&quot;, &quot;AAA&quot;, &quot;AAG&quot;,
                &quot;GAT&quot;, &quot;GAC&quot;, &quot;GAA&quot;, &quot;GAG&quot;,
                &quot;TGT&quot;, &quot;TGC&quot;, &quot;TGA&quot;, &quot;TGG&quot;,
                &quot;CGT&quot;, &quot;CGC&quot;, &quot;CGA&quot;, &quot;CGG&quot;,
                &quot;AGT&quot;, &quot;AGC&quot;, &quot;AGA&quot;, &quot;AGG&quot;,
                &quot;GGT&quot;, &quot;GGC&quot;, &quot;GGA&quot;, &quot;GGG&quot;
                )
    return(codons)
}


##' .. content for \description{} (estimate codon usage table based on training sequences) ..
##'
##' .. content for \details{} ..
##' @title getCodonUsage
##' @param sequences : A vector of sequence set.
##' @return A data.frame containing two columns of codons and Freq (frequency probabilities)
##' @author Guangchuang Yu
getCodonUsage &lt;- function(sequences) {
    codons &lt;- getCodon()
    codon.table &lt;- data.frame(codons=codons, Freq=rep(0, length(codons)))
    rownames(codon.table) = codons

    for (i in 1:length(sequences)) {
        codon &lt;- codonCount(sequences[i])
        rowname.idx &lt;- as.character(codon[,1])
        codon.table[rowname.idx,2] &lt;- codon.table[rowname.idx,2] + codon[,2]
    }
    codon.table[,2] &lt;- codon.table[,2]/sum(codon.table[,2])
                                        #print(codon.table)  #### print result.
    return(codon.table)
}

##' .. content for \description{} (no empty lines) ..
##'
##' .. content for \details{} ..
##' @title codonCount
##' @param seqStr : A sequence string
##' @return A data.frame containing two columns of codons and their counts
##' @author Guangchuang Yu
codonCount &lt;- function(seqStr) {
    codonLength=3
    len &lt;- nchar(seqStr)
    ## drop the last alphabets which is too short (

    idx &lt;- seq(1, len,codonLength)
    codons &lt;- c()
    for ( i in 1:length(idx)) {
        codon &lt;- substr(seqStr, idx[i], idx[i]+codonLength-1)
        codons &lt;- c(codons, codon)
    }
    codons.df &lt;- as.data.frame(table(codons))
    return(codons.df)
}

##' .. content for \description{} (reverse complementary sequence) ..
##'
##' .. content for \details{} ..
##' @title revcom
##' @param seqStr : A sequence string
##' @return A reverse complementary sequence
##' @author Guangchuang Yu
revcom &lt;- function(seqStr) {
    seqStr.str &lt;- unlist(strsplit(seqStr, split=&quot;&quot;))
    idxT &lt;- which(seqStr.str==&quot;T&quot;)
    idxG &lt;- which(seqStr.str==&quot;G&quot;)
    idxA &lt;- which(seqStr.str==&quot;A&quot;)
    idxC &lt;- which(seqStr.str==&quot;C&quot;)
    seqStr.str[idxT] &lt;- &quot;A&quot;
    seqStr.str[idxA] &lt;- &quot;T&quot;
    seqStr.str[idxG] &lt;- &quot;C&quot;
    seqStr.str[idxC] &lt;- &quot;G&quot;
    seqStr.rc &lt;- rev(seqStr.str)
    seqStr.rc &lt;- paste(seqStr.rc, collapse=&quot;&quot;)
    return(seqStr.rc)
}

##' .. content for \description{} (calculate NT frequency probabilities of a given sequence set) ..
##'
##' .. content for \details{use to estimate the background/random NT usage} ..
##' @title getNtUsage
##' @param file : A FASTA file
##' @return A data.frame containing NTs and their corresponding frequency probabilities.
##' @author Guangchuang Yu
getNtUsage &lt;- function(file) {
    print(&quot;reading FASTA file&quot;)
    fasta &lt;- readFASTA(file)
    sequences &lt;- sapply(fasta, function(i) i$seq)
    x &lt;- sapply(sequences, strsplit, split=&quot;&quot;)
    x &lt;- unlist(x)
    NTprob &lt;- as.data.frame(table(x))
    NTprob[,2] &lt;- NTprob[,2]/sum(NTprob[,2])
    rownames(NTprob) &lt;- NTprob[,1]
    return(NTprob)
}

##' .. content for \description{} (calculate the log likelihood of the six possible reading frame of a given sequence) ..
##'
##' .. content for \details{} ..
##' @title logLikelihood
##' @param sequence : A sequence string
##' @param transition : Transition Probatility Matrix
##' @param codonusage : Codon Usage data.frame
##' @param background : NT Usage data.frame for background/random model
##' @return Log likelihood
##' @author Guangchuang Yu
logLikelihood &lt;- function(sequence, transition, codonusage, background) {
    codonLength &lt;- 3
    orf1 &lt;- sequence
    orf2 &lt;- substr(sequence, 2, nchar(sequence))
    orf3 &lt;- substr(sequence, 3, nchar(sequence))
    seq.rc &lt;- revcom(sequence)
    orf4 &lt;- seq.rc
    orf5 &lt;- substr(seq.rc, 2, nchar(seq.rc))
    orf6 &lt;- substr(seq.rc, 3, nchar(seq.rc))
    ORF &lt;- c(orf1, orf2, orf3, orf4, orf5, orf6)
    names(ORF) &lt;- c(&quot;ORF1&quot;, &quot;ORF2&quot;, &quot;ORF3&quot;, &quot;ORF4&quot;, &quot;ORF5&quot;, &quot;ORF6&quot;)

    result &lt;- data.frame(ORF=names(ORF), logLikelihood=rep(0,6))
    for (j in seq_along(ORF)) {
        firstCodon &lt;- substr(ORF[j], 1, 3)
        if (firstCodon %in% codonusage[,1]) {
            initProb &lt;- log(codonusage[firstCodon,2])
        } else {
            initProb &lt;- log(1/64)
        }

        transitionCount &lt;- matrix(0, nrow=nrow(transition), ncol=ncol(transition))
        rownames(transitionCount) &lt;- rownames(transition)
        colnames(transitionCount) &lt;- colnames(transition)

        codons &lt;- rownames(transition)

        len &lt;- nchar(ORF[j])
        len &lt;- len - len %% codonLength - codonLength

        idx &lt;- seq(1, len, codonLength)
        for (i in seq_along(idx)) {
            ## codonI--codonJ
            codonI &lt;- substr(ORF[j],idx[i], idx[i]+codonLength-1)
            if (i != length(idx)) {
                codonJ &lt;- substr(ORF[j],idx[i+1], idx[i+1]+codonLength-1)
            }
            if (codonI %in% codons &amp; codonJ %in% codons) {
                transitionCount[codonI, codonJ] &lt;- transitionCount[codonI, codonJ] +1
            } else {
                warning(&quot;codon &quot;, codonI, &quot; or &quot;, codonJ, &quot; which does not defined was detected...&quot;)
            }
        }

        postTransition &lt;- log(transition) * transitionCount
        posterior &lt;- initProb + sum(postTransition)

        NTseq &lt;- unlist(strsplit(ORF[j], split=&quot;&quot;))
        orf.len &lt;- nchar(ORF[j]) - nchar(ORF[j]) %% codonLength
        NTseq &lt;- NTseq[1:orf.len]
        randomModel &lt;- sum(log(background[NTseq,2]))

        likelihood &lt;- posterior - randomModel
        result[j, 2] &lt;- likelihood
    }
    return(result)
}
</code></pre>
              
                <center><span class="zaln"><a href="https://mp.weixin.qq.com/s?__biz=MzI5NjUyNzkxMg==&mid=2247484170&idx=1&sn=ce5f7eb6df66d79661276a64bf25b4bc&chksm=ec43b24ddb343b5b2877574153aeccd049a6824716063883edb35c2338bd32c067caf3885f53#rd" class="zb">èµèµ</a></span></center>
                <center><span class="zaln"><a href="https://guangchuangyu.github.io/cn/2017/11/bmc-journal-protects-paper-of-plagiarism/" class="zb">æŠµåˆ¶BMCï¼ŒåŒ¹å¤«æœ‰è´£</a></span></center>
            </div>
          </div>
          <div id="post-footer" class="post-footer main-content-wrap">
            
              
                
                
                  <div class="post-footer-tags">
                    <span class="text-color-light text-small">TAGGED IN</span><br/>
                    
  <a class="tag tag--primary tag--small" href="https://guangchuangyu.github.io/cn//tags/ibw/">IBW</a>

                  </div>
                
              
            
            <div class="post-actions-wrap">
  <nav>
    <ul class="post-actions post-action-nav">
      
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/07/ibw2011/" data-tooltip="IBW2011">
          
            <i class="fa fa-angle-left"></i>
            <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
          </a>
        </li>
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/06/screen-shot/" data-tooltip="screen shot">
          
            <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
            <i class="fa fa-angle-right"></i>
          </a>
        </li>
      
    </ul>
  </nav>
  <ul class="post-actions post-action-share">
    
      <li class="post-action hide-lg hide-md hide-sm">
        <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
          <i class="fa fa-share-alt"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-google-plus"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-facebook-official"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-twitter"></i>
        </a>
      </li>
    
    
      <li class="post-action">
        <a class="post-action-btn btn btn--default" href="#disqus_thread">
          <i class="fa fa-comment-o"></i>
        </a>
      </li>
    
    <li class="post-action">
      
        <a class="post-action-btn btn btn--default" href="#">
      
        <i class="fa fa-list"></i>
      </a>
    </li>
  </ul>
</div>


            
              
                <div id="disqus_thread">
  <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
              
            
          </div>
        </article>
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2017 Guangchuang Yu. All Rights Reserved
  </span>
</footer>

      </div>
      <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
        <div class="post-actions-wrap">
  <nav>
    <ul class="post-actions post-action-nav">
      
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/07/ibw2011/" data-tooltip="IBW2011">
          
            <i class="fa fa-angle-left"></i>
            <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
          </a>
        </li>
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/06/screen-shot/" data-tooltip="screen shot">
          
            <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
            <i class="fa fa-angle-right"></i>
          </a>
        </li>
      
    </ul>
  </nav>
  <ul class="post-actions post-action-share">
    
      <li class="post-action hide-lg hide-md hide-sm">
        <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
          <i class="fa fa-share-alt"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-google-plus"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-facebook-official"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-twitter"></i>
        </a>
      </li>
    
    
      <li class="post-action">
        <a class="post-action-btn btn btn--default" href="#disqus_thread">
          <i class="fa fa-comment-o"></i>
        </a>
      </li>
    
    <li class="post-action">
      
        <a class="post-action-btn btn btn--default" href="#">
      
        <i class="fa fa-list"></i>
      </a>
    </li>
  </ul>
</div>


      </div>
      <div id="share-options-bar" class="share-options-bar" data-behavior="4">
  <ul class="share-options">
    <li class="share-option">
      <a class="share-option-btn" target="new" href="https://plus.google.com/share?url=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
        <i class="fa fa-google-plus"></i><span>Share on Google Plus</span>
      </a>
    </li>
    <li class="share-option">
      <a class="share-option-btn" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
        <i class="fa fa-facebook-official"></i><span>Share on Facebook</span>
      </a>
    </li>
    <li class="share-option">
      <a class="share-option-btn" target="new" href="https://twitter.com/intent/tweet?text=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
        <i class="fa fa-twitter"></i><span>Share on Twitter</span>
      </a>
    </li>
  </ul>
</div>
<div id="share-options-mask" class="share-options-mask"></div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Guangchuang Yu</h4>
    
      <div id="about-card-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Postdoc researcher
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        Hong Kong
      </div>
    
  </div>
</div>

    <div id="algolia-search-modal" class="modal-container">
  <div class="modal">
    <div class="modal-header">
      <span class="close-button"><i class="fa fa-close"></i></span>
      <a href="https://algolia.com" target="_blank" class="searchby-algolia text-color-light link-unstyled">
        <span class="searchby-algolia-text text-color-light text-small">by</span>
        <img class="searchby-algolia-logo" src="https://www.algolia.com/static_assets/images/press/downloads/algolia-light.svg">
      </a>
      <i class="search-icon fa fa-search"></i>
      <form id="algolia-search-form">
        <input type="text" id="algolia-search-input" name="search"
          class="form-control input--large search-input" placeholder="Search" />
      </form>
    </div>
    <div class="modal-body">
      <div class="no-result text-color-light text-center">no post found</div>
      <div class="results">
        
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/ggimage-hat/">
                <h3 class="media-heading">ç»™ä½ ä¸€é¡¶å¸½å­</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>åœ£è¯èŠ‚æœŸé—´ï¼Œä¸€ä¸ªä¸ªåœ¨@å¾®ä¿¡å®˜æ–¹ï¼Œè¦åŠ åœ£è¯å¸½ï¼Œè¿™è®©äººæƒ³èµ·äº†ä»¥å‰ï¼š</p>

<blockquote>
<p>ä»Šå¤©æ˜¯é©¬åŒ–è…¾çš„ç”Ÿæ—¥ï¼Œè½¬å‘5ä¸ªç¾¤ï¼Œè‡ªåŠ¨å¼€é€šä¸€ä¸ªæœˆçš„QQä¼šå‘˜</p>
</blockquote>

<p>è¿™ç­ç©QQé•¿å¤§çš„å­©å­ï¼Œç°åœ¨å˜æˆäº†ç©å¾®ä¿¡çš„æ²¹è…»ä¸­å¹´äº†ã€‚</p>

<p>åšä¸ºbiobabbleå…¬ä¼—å·çš„ç²‰ä¸ï¼Œæˆ‘è§‰å¾—ä½ ä»¬æ˜¯æœ‰èƒ½åŠ›ç”¨ä¸¤ä¸‰è¡ŒRä»£ç æ¥ç»™è‡ªå·±åŠ ä¸ªå¸½å­çš„ã€‚ç”±äºæˆ‘åœ¨è¿‡åœ£è¯èŠ‚ï¼Œæ‰€ä»¥è¿™ç¯‡æ–‡ç« æ²¡æœ‰èƒ½å¤ŸåŠæ—¶æ¨é€å‡ºæ¥ï¼Œå°±å½“é©¬åç‚®å§ï¼Œä¸åˆæ—¶å®œåœ°è°ƒä¾ƒä¸€ä¸‹å¤§å®¶ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/biosyntax/">
                <h3 class="media-heading">bioSyntax - é’ˆå¯¹ç”Ÿç‰©æ•°æ®çš„è¯­æ³•é«˜äº®</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p><img src="http://biosyntax.org/images/sam-less-small.gif" alt="" /></p>

<p>å¯¹äºç”Ÿä¿¡ç‹—ï¼Œå¤©å¤©é»‘ç™½å‘½ä»¤è¡Œï¼Œå®éªŒå®¤çš„åŒäº‹ï¼Œç»å¸¸å¯¹ç€æˆ‘çš„Emacsè¯´ï¼Œè¿™ä¹ˆå°çš„å­—ä½“ä½ çœ‹å¾—æ¸…å—ï¼Ÿæˆ‘åªèƒ½æ— å¥ˆåœ°å›ç­”ï¼Œå› ä¸ºæˆ‘è¦ä¸€ä¸ªå±å¹•æ˜¾ç¤ºä¸€æ•´ä¸ªå‡½æ•°ã€‚æˆ‘è‡ªå·±å¹³æ—¶å†™ä»£ç ä¹Ÿå¾ˆæ³¨æ„è¿™ä¸€ç‚¹ï¼Œå‡½æ•°å°½é‡å°ï¼Œå¤ªé•¿å°±è¦è€ƒè™‘åˆ‡åˆ†æˆå‡ ä¸ªå‡½æ•°ï¼Œæˆ‘è¯´è‡ªå·±æ˜¯è„‘å®¹é‡ä¸å¤Ÿï¼Œå†™ä¸äº†é•¿å‡½æ•°ã€‚æˆ‘å¸Œæœ›å¦‚æœæœ‰äººçœ‹æˆ‘ä»£ç ï¼Œä»–/å¥¹å¯ä»¥çœ‹å¾—èˆ’æœç‚¹ï¼Œè™½ç„¶å†™çš„ä¸è§å¾—å¥½ï¼Œä½†èµ·ç ä¸€ä¸ªå‡½æ•°ï¼Œä½ ä¸€ä¸ªå±å¹•æ˜¯è£…å¾—ä¸‹çš„ã€‚å²æœˆç•™ç»™æˆ‘ä»¬çš„ï¼Œæ˜¯è¶Šæ¥è¶Šåšçš„é•œç‰‡ï¼ç„¶è€Œæœ€è¿‘æœ‰ä¸€æ¬¾ç¥å™¨ï¼Œè®©æˆ‘ä»¬åœ¨å‘½ä»¤è¡Œé‡Œï¼Œå¯ä»¥é«˜äº®æ˜¾ç¤ºç”Ÿç‰©å­¦å¸¸è§çš„æ•°æ®æ ¼å¼ï¼ŒåŒ…æ‹¬SAM, VCF, GTF, PDBå’ŒFASTAã€‚è¿™æ¬¾ç¥å™¨å«bioSyntax (<a href="http://bioSyntax.org">http://bioSyntax.org</a>) ï¼Œçœ‹ç€èˆ’æœå¤šäº†ï¼Œè€Œä¸”é€¼æ ¼åè¶³ï¼Œå¦‚æœå¤§å®¶åºåˆ—æ¯”å¯¹çœ‹å¤šäº†ï¼Œå…¶å®éƒ½ä¼šä¾èµ–äºçœ‹é¢œè‰²ï¼Œé»‘ç™½å­—ä½“æƒ…å†µä¸‹ï¼Œä½ å¾ˆéš¾å»åŒºåˆ†ATCGï¼Œå³ä½¿æœ‰æ˜æ˜¾çš„patternï¼Œä½ ä¹Ÿå¾ˆéš¾çœ‹å‡ºæ¥ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/geom-cake/">
                <h3 class="media-heading">130å²äº†ï¼Œç¥ä½ ç”Ÿå¿«</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>å‘¨æ—¥æ˜¯æ¸¯å¤§åŒ»å­¦é™¢çš„198å±Šæ¯•ä¸šå…¸ç¤¼ï¼ŒYå”è¦å»å‚åŠ ï¼Œä»Šå¹´ä¹Ÿæ­£å¥½æ˜¯åŒ»å­¦é™¢130å‘¨å¹´ï¼Œåšä¸ºç”»å›¾ç•Œçš„ä¼ è¯´ï¼Œæˆ‘å†³å®šç”»ä¸€ä¸ªè›‹ç³•ã€‚</p>

<p>è¦ç”»è›‹ç³•ï¼Œè¿™æœ‰ä½•éš¾ï¼Œæˆ‘ä»¬æœ‰<code>emojifont</code>å•Šï¼Œè¿˜è®°å¾—ã€Š<a href="http://mp.weixin.qq.com/s/ARxyTRpMs6OeCqOQPp9OwA">emojifontæ–°åŠŸèƒ½ï¼ˆæœ‰è§†é¢‘å“¦ï¼‰</a>ã€‹ä¸€æ–‡ä¸­çš„<code>geom_emoji</code>å›¾å±‚å—ï¼Ÿ</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/chipseeker-8g/">
                <h3 class="media-heading">æœ€åçš„å…«å¦</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><h2 id="chipseekerç³»åˆ—ä¼ é€é—¨">ChIPseekerç³»åˆ—ä¼ é€é—¨</h2>

<ul>
<li><a href="https://mp.weixin.qq.com/s/lFi1m7k-9t6nD0SER1r1hw">CS0: ChIPseqä»å…¥é—¨åˆ°æ”¾å¼ƒ</a></li>
<li><a href="http://mp.weixin.qq.com/s/vgXrscsUkfYdC5qt_PUNlw">CS1: ChIPseqç®€ä»‹</a></li>
<li><a href="https://mp.weixin.qq.com/s/D06NEG7blksGugb0srL7KQ">CS2: BEDæ–‡ä»¶</a></li>
<li><a href="http://mp.weixin.qq.com/s/vWTf59KDs1lp_sQXjEhI_g">CS3: peakæ³¨é‡Š</a></li>
<li><a href="http://mp.weixin.qq.com/s/_OPzvaEAbiMolCA2mqJXLw">CS4ï¼šå…³äºChIPseqæ³¨é‡Šçš„å‡ ä¸ªé—®é¢˜</a></li>
<li><a href="http://mp.weixin.qq.com/s/gDpsdj6cLDtnG1Kla79wLA">CS5: åƒç€ç«é”…ï¼Œå”±ç€æ­Œï¼Œè¿˜æŠŠåˆ†æç»™åšäº†</a></li>
<li><a href="http://mp.weixin.qq.com/s/MqpfgkMJSFj0pYwcEjV9kQ">CS6: ChIPseekerçš„å¯è§†åŒ–æ–¹æ³•ï¼ˆä¸­ç§‹èŠ‚çš„è§†è§‰é¥•é¤®ï¼‰</a></li>
<li><a href="http://mp.weixin.qq.com/s/v61LjQEfyi_CZucHW53ETw">CS7ï¼šGenomic coordinationçš„å¯Œé›†æ€§åˆ†æï¼ˆ1ï¼‰</a></li>
<li><a href="https://mp.weixin.qq.com/s/vbkZrGwD7dZ-OPOiqkyTnA">CS8ï¼šGenomic coordinationçš„å¯Œé›†æ€§åˆ†æï¼ˆ2ï¼‰</a></li>
<li><a href="https://mp.weixin.qq.com/s/98k_L3CmjH817xMF2VfYtA">CS9: GEOæ•°æ®æŒ–æ˜</a></li>
</ul>

<p>ChIPseekerè¿™ä¸ªç³»åˆ—ï¼Œä»ã€Š<a href="https://mp.weixin.qq.com/s/lFi1m7k-9t6nD0SER1r1hw">CS0: ChIPseqä»å…¥é—¨åˆ°æ”¾å¼ƒ</a>ã€‹åˆ°ç°åœ¨CS10ï¼Œæ€»å…±11ç¯‡ï¼Œç¬¬ä¸€ç¯‡ä»¥å…«å¦å¼€å§‹ï¼Œæˆ‘æƒ³å¾ˆæœ‰å¿…è¦ä»¥å…«å¦æ¥ç»“æŸã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/geo-data-mining-for-chipseq/">
                <h3 class="media-heading">ChIPseq-GEOæ•°æ®æŒ–æ˜</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>ChIPseekerç³»åˆ—æ–‡ç« å·²ç»ä»‹ç»äº†å¾ˆå¤šå†…å®¹ï¼ŒåŒ…æ‹¬æ³¨é‡Šçš„æ–¹æ–¹é¢é¢ï¼Œä¹ŸåŒ…æ‹¬å¼ºå¤§çš„å¯è§†åŒ–åŠŸèƒ½ï¼ˆã€Š<a href="http://mp.weixin.qq.com/s/MqpfgkMJSFj0pYwcEjV9kQ">CS6: ChIPseekerçš„å¯è§†åŒ–æ–¹æ³•ï¼ˆä¸­ç§‹èŠ‚çš„è§†è§‰é¥•é¤®ï¼‰</a>ã€‹ï¼‰ã€‚</p>

<p>ä»Šå¤©è¦ä»‹ç»ä¸€ä¸‹æ•°æ®æŒ–æ˜ï¼Œä»å¤§é‡å·²æœ‰çš„æ•°æ®æ¥äº§ç”Ÿæ–°çš„hypothesisã€‚æ­£å¦‚æˆ‘åœ¨ChIPseekerçš„æ–‡ç« é‡Œå†™çš„ï¼š</p>

<blockquote>
<p>There are increasing evidences shown that combinations of TFs are important for regulating gene expression (Perez-Pinera et al., 2013; Zhu et al., 2008). However, systematically identification of TF interactions by ChIP-seq is still not available. Even if a specific TF binding is essential for a particular regulation was known, we do not have prior knowledge of all its co-factors. There are no systematic strategies available to identified un-known co-factors by ChIP- seq.</p>
</blockquote>

<p>å¹¶æ²¡æœ‰æ–¹æ³•å¯ä»¥å¤§è§„æ¨¡åœ°é¢„æµ‹æœªçŸ¥çš„å…±åŒè°ƒæ§å› å­ï¼Œè€Œæ•°æ®æŒ–æ˜å°±æ˜¯è¦ç»™æˆ‘ä»¬è¿™ç§é¢„æµ‹çš„èƒ½åŠ›ã€‚</p>

<p>æˆ‘å½“å¹´åœ¨å†™ChIPseekerçš„æ—¶å€™ï¼Œæˆ‘æœ‰çº ç»“æ˜¯å†™ç¯‡Bioinformaticsçš„application noteå‘¢ï¼Œè¿˜æ˜¯å†™ç¯‡é•¿æ–‡çŒæ°´NARï¼Œæ¯•ç«ŸNARå½±å“å› å­é«˜ä¸€ç‚¹ï¼Œæœ€åè¿˜æ˜¯å‘äº†Bioinformaticsï¼Œå› ä¸ºæˆ‘æ²¡é’±ï¼Œå›§ï¼ŒBioinformaticsä¸è¦ç‰ˆé¢è´¹å•Šã€‚ç„¶åé™äºç¯‡å¹…ï¼ŒChIPseekeræœ‰å¤§é‡å¯è§†åŒ–çš„å‡½æ•°ï¼Œæˆ‘åœ¨æ–‡ç« ä¸­ä¸€å¼ å›¾éƒ½æ²¡æ”¾ï¼ï¼ï¼å¦‚æœå½“æ—¶å†³å®šå‘NARçš„è¯ï¼Œè¿™ä¸ªæ•°æ®æŒ–æ˜è¿™ä¸€å—æˆ‘å°±ä¼šå†™å¤šç‚¹ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/using-external-font-for-meme/">
                <h3 class="media-heading">ä½¿ç”¨å¤–éƒ¨å­—ä½“ç”»meme</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>memeæ— éæ˜¯ç…§ç‰‡ä¸ŠåŠ ä¸¤è¡Œå­—ï¼Œ<a href="https://cran.r-project.org/web/packages/meme/">meme</a>è¿™ä¸ªåŒ…æœ¬æ¥æ˜¯ç»ƒæ‰‹ä¹‹ä½œï¼Œç”¨æ¥<a href="https://guangchuangyu.github.io/2017/10/create-meme-in-r/">æ¨¡æ‹Ÿggplot2çš„api</a>ã€‚ç»“æœæˆ‘åé¢åˆå®ç°äº†<a href="https://guangchuangyu.github.io/2017/11/shadow-text-effect-in-grid-and-ggplot2-graphics/">å­—ä½“é˜´å½±æ•ˆæœ</a>ï¼Œè¿™æ¬¡æˆ‘åˆå¸¦æ¥äº†å¤–éƒ¨å­—ä½“æ”¯æŒã€‚</p>

<p>åŠ è½½å¤–éƒ¨å­—ä½“åœ¨Rä¸Šé¢ä¸»è¦æœ‰extrafontå’Œshowtextä¸¤ä¸ªå®ç°ï¼Œç‰¹åˆ«æ˜¯showtextç”¨èµ·æ¥è¿˜æ˜¯éå¸¸ç®€å•çš„ã€‚æˆ‘åœ¨memeåŒ…é‡Œå†™äº†<code>font_import</code>å‡½æ•°ï¼Œå†…éƒ¨å°±æ˜¯ç”¨showtextï¼Œåªæ˜¯ç®€å•çš„åŒ…è£…äº†ä¸€ä¸‹ï¼Œè¿™æ ·å°ç™½ç”¨æˆ·å¯ä»¥ç›´æ¥ä¸‹ä¸ªå­—ä½“å°±å¼€æ’¸ã€‚æœ‰é˜´å½±æ•ˆæœï¼Œæœ‰<a href="https://fontmeme.com/fonts/">æç¬‘å­—ä½“</a>ï¼Œå°±ä¸‡äº‹å…·å¤‡äº†ï¼Œè¿™ä¿¨ç„¶æ˜¯ä¸ªä¸“ä¸šçš„memeå·¥å…·äº†ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/genomic-coordination-enrichment-analysis/">
                <h3 class="media-heading">Genomic coordinationçš„å¯Œé›†æ€§åˆ†æ</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>åœ¨ã€Š<a href="https://mp.weixin.qq.com/s/v61LjQEfyi_CZucHW53ETw">CS7ï¼šGenomic coordinationçš„å¯Œé›†æ€§åˆ†æï¼ˆ1ï¼‰</a>ã€‹è¯´åˆ°äº†seq2pathwayè¿™ä¸ªåŒ…ï¼Œå…¶å®æ˜¯ä¸¤éƒ¨æ›²ï¼Œseq2gene-&gt;gene2pathwayï¼Œæ— éæ˜¯æŠŠæµ‹åºç‰‡æ®µç”¨ä¸´è¿‘çš„åŸºå› æ³¨é‡Šï¼ŒåŒ…æ‹¬å’ŒTSS overlapçš„åŸºå› ï¼Œå®¿ä¸»åŸºå› ï¼Œä¸Šä¸‹æ¸¸çš„åŸºå› ç­‰ï¼Œç„¶åæ‹¿è¿™äº›åŸºå› è·‘ORAï¼Œåšå¯Œé›†ï¼Œä»…æ­¤è€Œå·²ï¼Œè¿™ä¸ªåŒ…æ”¯æŒçš„ç‰©ç§ææœ‰é™ï¼Œã€Š<a href="http://mp.weixin.qq.com/s/_OPzvaEAbiMolCA2mqJXLw">CS4ï¼šå…³äºChIPseqæ³¨é‡Šçš„å‡ ä¸ªé—®é¢˜</a>ã€‹è¿™ä¸€æ–‡ä¸­è®²åˆ°ChIPseekeræ”¯æŒæ‰€æœ‰æœ‰åŸºå› ç»„æ³¨é‡Šçš„ç‰©ç§ï¼Œè€Œã€Š<a href="http://mp.weixin.qq.com/s/ScwaofLd_ybfECYjL1dGow">clusterProfiler for enrichment analysis</a>ã€‹ä¹Ÿæ”¯æŒæ‰€æœ‰ç‰©ç§ï¼ˆå³ä½¿ä½ è‡ªå·±è·‘çš„ç”µå­æ³¨é‡Šï¼Œä¹Ÿèƒ½æ”¯æŒï¼‰ï¼Œé‚£ä¹ˆä½¿ç”¨ChIPseekeræ¥åšåŸºå› æ³¨é‡Šï¼Œç„¶åè¡”æ¥clusterProfilerå°±å¯ä»¥æ”¯æŒæ‰€æœ‰ç‰©ç§çš„æµ‹åºç‰‡æ®µè¿›è¡ŒåŠŸèƒ½å¯Œé›†åˆ†æäº†ã€‚</p>

<p>ã€Š<a href="http://mp.weixin.qq.com/s/vWTf59KDs1lp_sQXjEhI_g">CS3: peakæ³¨é‡Š</a>ã€‹æœ¬èº«å°±æ”¯æŒå‡ ç§æ³¨é‡Šï¼Œå¦å¤–æˆ‘å†™äº†ä¸€ä¸ª<code>seq2gene</code>çš„å‡½æ•°ï¼Œå¥—ç”¨<code>seq2pathway</code>çš„æ€è·¯ï¼ŒæŠŠä¸€ä¸ªåŸºå› ä½ç½®ä¸Šæ‰€æœ‰å…³è”çš„åŸºå› å…¨éƒ¨è¿”å›æ¥ï¼Œæˆ‘ä»¬å¯ä»¥ä½¿ç”¨å®ƒå»æŠŠåŸºå› ä½ç½®ä¿¡æ¯è½¬æ¢æˆåŸºå› åˆ—è¡¨ï¼Œç„¶åç”¨äºå¯Œé›†åˆ†æï¼Œè¿˜æ˜¯ç†Ÿæ‚‰çš„å‘³é“ï¼Œè¿˜æ˜¯ç†Ÿæ‚‰çš„é…æ–¹ğŸ¦„</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/12/subsetting-in-ggplot2-ggtree/">
                <h3 class="media-heading">ggplot2/ggtreeå–å­é›†ç”»å›¾</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Dec 12, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>å–å­é›†å¯¹äºè¿›åŒ–æ ‘å¯è§†åŒ–æ¥è¯´æ˜¯éå¸¸å¸¸è§çš„ï¼Œæˆ‘ä»¬è¦åŒºåˆ†å†…éƒ¨èŠ‚ç‚¹å’Œå¤–éƒ¨èŠ‚ç‚¹ï¼Œæˆ‘ä»¬ä¹Ÿå¯èƒ½æƒ³é’ˆå¯¹æŸäº›ç‰¹ç‚¹çš„èŠ‚ç‚¹è¿›è¡Œæ³¨é‡Šã€‚</p>

<p>ggplot2ç°åœ¨æ‰€æœ‰å›¾å±‚éƒ½ä¸æ”¯æŒç›´æ¥å–å­é›†ï¼Œæ‰€ä»¥å‘¢ggtreeå°±è‡ªå·±å®šä¹‰äº†ä¸€äº›ä¿®æ”¹çš„å›¾å±‚ï¼ŒåŒ…æ‹¬<code>geom_text2</code>, <code>geom_label2</code>,  <code>geom_point2</code> å’Œ <code>geom_segment2</code>ï¼Œè¿™äº›å›¾å±‚å’Œggplot2çš„ç‰ˆæœ¬å”¯ä¸€å·®åˆ«å°±æ˜¯æ”¯æŒå–å­é›†ã€‚è¿™æ ·å¯¹äºæˆ‘ä»¬åšæ³¨é‡Šæ¥è¯´ï¼Œå°±æ›´æ–¹ä¾¿äº†ã€‚</p>

<p>æ¯”å¦‚è¯´æˆ‘æƒ³ç»™å†…éƒ¨èŠ‚ç‚¹æ‰“ç‚¹ï¼Œå¯ä»¥ç”¨ï¼š</p>

<pre><code class="language-r">ggtree(tree) + geom_text2(aes(label=label, subset=!isTip), hjust=-.2) +
        geom_point2(aes(subset=!isTip), color=&quot;red&quot;, size=3)
</code></pre>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/11/sci-hub-tor/">
                <h3 class="media-heading">å†ä¸ç”¨æ‹…å¿ƒæ‰“ä¸å¼€sci-hubäº†</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Nov 11, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>ä¹‹å‰æœ‰å†™è¿‡<a href="https://mp.weixin.qq.com/s/AGQn7Lfn_nHHrtVQyJSTmg">è¯»æ–‡çŒ®çš„æ–‡ç« </a>ï¼Œä¹Ÿä»‹ç»äº†<a href="https://mp.weixin.qq.com/s/TMmsOsEn0wV9yXKOLma3Rw">å¦‚ä½•è·å–æ–‡çŒ®</a>ï¼Œé‚£æ˜¯æˆ‘è¿˜åœ¨è¯»ç¡•å£«çš„æ—¶å€™å†™çš„ï¼Œè¦†ç›–äº†æ‰¾æ–‡çŒ®ã€æ–‡çŒ®ç®¡ç†ã€ä¸‹è½½æ–‡çŒ®ã€è®¢é˜…æœŸåˆŠç­‰æ–¹æ–¹é¢é¢ï¼Œè¯´åˆ°ä¸‹æ–‡çŒ®ï¼Œæœ€å¤§çš„ç¥å™¨å½“ç„¶æ˜¯sci-hubï¼Œä½†æ˜¯sci-hubçš„åŸŸåsci-hub.cc, sci-hub.bz, sci-hub.ioçº·çº·éƒ½æ²¦é™·ï¼Œæœ€è¿‘å›½å†…æœ‰ä¸€ç¥å™¨geenmedicalï¼Œæ•´åˆäº†å½±å“å› å­ã€å¼•ç”¨æ•°ç­‰æŒ‡æ ‡ï¼Œç„¶è€Œè¿™ä¸€ç¥å™¨çš„ä¸‹æ–‡çŒ®åŠŸèƒ½åªæ˜¯æä¾›å¤–é“¾ï¼Œæœ‰ä¸€ç§è¯´æ³•æ˜¯geenmedical=pubmed+scihubï¼Œå½“ç„¶geenmedicalçš„èµ„æºè¿˜å¯èƒ½æ¥è‡ªäºç™¾åº¦å­¦æœ¯ã€researchGateç­‰ï¼Œä¸éƒ½æ˜¯æ¥è‡ªäºsci-hubï¼Œç„¶è€Œæˆ‘ä»¬å¿…é¡»æ‰¿è®¤çš„æ˜¯æœ€å¤§çš„æ–‡çŒ®æä¾›æ¥æºå¿…é¡»è¿˜å¾—æ˜¯sci-hubï¼Œæ‰€ä»¥å¦‚æœsci-hubæŒ‚äº†ï¼ŒåŸºäºsci-hubçš„ç¥å™¨ä¹Ÿå°±ç¥å¥‡ä¸èµ·æ¥äº†ã€‚å¦‚æœæœ‰äººè·Ÿä½ è¯´å¿˜è®°sci-hubå§ï¼Œç”¨geenmedicalï¼Œé‚£ç»å¯¹æ˜¯æ ‡é¢˜å…šï¼è¿™é‡Œå¹¶æ²¡æœ‰è¯´geenmedicalä¸å¥½ï¼Œå®ƒæ•´åˆäº†å¤šæ–¹é¢çš„èµ„æºå¹¶ä¸”æœ¬åœ°åŒ–å¾—éå¸¸å¥½ï¼Œå¾ˆé€‚åˆå›½å†…ä½¿ç”¨ã€‚</p>

<p>æ¯”å¦‚æˆ‘åœ¨geenmedicalé‡ŒæœclusterProfilerï¼Œç»™å‡ºçš„ä¸‹è½½é“¾æ¥ç¬¬ä¸€ä¸ªå°±æ˜¯sci-hubçš„ï¼š<a href="http://doi.org.sci-hub.cc/10.1089/omi.2011.0118">http://doi.org.sci-hub.cc/10.1089/omi.2011.0118</a>ï¼Œç„¶è€Œè¿™ä¸ªé“¾æ¥æ˜¯ä¸å¯ç”¨çš„ï¼å½“ç„¶å®ƒç»™å‡ºäº†æ¯”å¦‚ç™¾åº¦å­¦æœ¯çš„é“¾æ¥æ˜¯å¯ä»¥ä¸‹è½½åˆ°æ–‡çŒ®çš„ã€‚ç„¶è€Œè¿˜æ˜¯é‚£å¥è¯ï¼Œè®ºã€Œæµ·ç›—ç²¾ç¥ã€æˆ‘åªæœsci-hubï¼Œè™½ç„¶æˆ‘ä¹‹å‰ä¹Ÿä»‹ç»äº†ã€Š<a href="http://mp.weixin.qq.com/s/aaIby87LSQgxpvCAECFxkw">sci-hubå¦‚æœæŒ‚äº†ï¼Œä½ è¿˜æœ‰ç¥å™¨ä¸‹æ–‡çŒ®å—ï¼Ÿ</a>ã€‹ï¼Œä½†ä»»ä½•ä¸‹æ–‡çŒ®çš„ç¥å™¨ï¼Œåœ¨sci-hubé¢å‰éƒ½æ˜¯æ¸£æ¸£ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2017/11/background-rect-for-selected-facet/">
                <h3 class="media-heading">ggplot2ç»™é€‰å®šçš„åˆ†é¢åŠ èƒŒæ™¯</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Nov 11, 2017
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>æˆ‘è¿™ä¸ªå…¬ä¼—å·ä¸åƒå¤§å¤šæ•°æ‰€è°“çš„ç”Ÿä¿¡ä»å…¥é—¨åˆ°ç²¾é€šçš„å„ç§å…¶å®åªæ˜¯æ¬è¿ç‚¹å…¥é—¨æ•™ç¨‹çš„ç¾¤ä¼—å–œé—»ä¹è§çš„å…¬ä¼—å·ã€‚æ­£å¦‚æˆ‘åœ¨ã€Š<a href="http://mp.weixin.qq.com/s/f-c5Q63SY166ZGzYCsqthA">ä¸ºä»€ä¹ˆè¦å¼€è¿™ä¸ªå…¬ä¼—å·</a>ã€‹é‡Œè¯´çš„ï¼Œè¿™æ˜¯å°ä¼—çš„ï¼Œæœ‰ä¸ªäººè‰²å½©çš„å„ç§åŸåˆ›æ–‡ã€‚æˆ‘ä¸å¯èƒ½åƒå…¶å®ƒå…¬ä¼—å·ä¸€æ ·æ‹›ä¸¤å°å¼Ÿå½“å®¢æœï¼Œå¾ˆå¤šäººåœ¨å…¬ä¼—å·åå°å‘æˆ‘æ‰”äº†è®¸å¤šé—®é¢˜ï¼Œç”±äºä¸ªäººç²¾åŠ›æœ‰é™ï¼Œåªèƒ½ä¼˜å…ˆè§£ç­”ã€ŒçŸ¥è¯†æ˜Ÿçƒã€çš„é—®é¢˜ï¼Œä¸Šæ¬¡å†™çš„ã€Š<a href="http://mp.weixin.qq.com/s/jegI1wWc7DJzrM7mQfz77Q">åŒä¸€æ•°æ®å¤šå˜é‡åˆ†ç»„çš„boxplot?</a>ã€‹ï¼Œå›¾è™½ç„¶ç®€å•ï¼Œå´ç©¿ç€å¥½å¤šä»¶é©¬ç”²ï¼Œè€Œæˆ‘æŠŠå®ƒæ‰’å…‰äº†ç»™ä½ çœ‹ğŸ™ˆ</p>

<p><img src="https://guangchuangyu.github.io/blog_images/zhishixingqiu/ggplot-bg_question.jpg" alt="" /></p>

<p>ä¸Šé¢è¿™ä¸ªå›¾ï¼Œä½ çœ‹ç€é«˜å¤§ä¸Šå§ï¼Œæˆ‘éƒ½å¯ä»¥åå®ƒä¸€è„¸å£æ°´ã€‚æ¯ä¸€ä¸ªæœ‰ç°è‰²èƒŒæ™¯çš„å›¾ï¼Œåœ¨xè½´ä¸Šviolinéƒ½å¤Ÿåˆ°è¾¹ç•Œäº†ï¼Œå…¶å®æ‰€æœ‰çš„violinéƒ½å¤Ÿåˆ°äº†ï¼Œè¿™è¯æ˜ä»€ä¹ˆï¼Ÿæ¯ä¸ªviolinä¹‹é—´å…¶å®ä¸å¯æ¯”è¾ƒï¼ä½ èƒ½æƒ³åƒå‡ ä¸ªç‹¬ç«‹çš„æ•°æ®ï¼Œåœ¨ç»Ÿä¸€çš„bin widthæƒ…å†µä¸‹ï¼Œç”»density curveï¼Œç«Ÿç„¶æœ€é«˜ç‚¹éƒ½ä¸€æ ·é«˜å—ï¼Ÿæ˜¾ç„¶å¯èƒ½æ€§å‡ ä¹ä¸º0ã€‚è¿™ä¸ªå¦‚æœä½¿ç”¨ggplot2çš„è¯ï¼Œå¯ä»¥ä½¿ç”¨<code>scale='width'</code>å¼ºåˆ¶æ‹‰æˆä¸€æ ·é«˜ï¼Œä½†æˆ‘ä¸æ¨èï¼Œæ­£å¦‚æˆ‘å‰é¢è¯´çš„ï¼Œä¸å¯æ¯”è¾ƒäº†ã€‚é»˜è®¤å‚æ•°<code>scale='area'</code>ï¼Œç§¯åˆ†é¢ç§¯ä¸€æ ·ï¼Œå’Œdensity curveä¸€æ ·è§£æï¼Œå¦å¤–çš„å‚æ•°<code>scale=&quot;count&quot;</code>ï¼Œé«˜åº¦ä¸è®¡æ•°åŒæ¯”ä¾‹ï¼Œå’Œhistogramä¸€æ ·è§£æï¼Œè€Œ<code>scale='width'</code>å¼ºåˆ¶æ‹‰æˆä¸€æ ·é«˜ï¼Œå¦‚æœæ²¡æœ‰åœ¨æ˜¾çœ¼å¤„è¯´æ˜ï¼Œè¯¯å¯¼æ€§å¤ªå¼ºã€‚</p>

<p>ç”»è¿™ç§å›¾ä¹Ÿå¯ä»¥æ‰‹å·¥æ‹¼ï¼Œè¿™æ ·å°±ç®€å•äº†ã€‚åœ¨ä½ éœ€è¦çš„æƒ…å†µä¸‹ï¼ŒåŠ ä¸ªç°è‰²èƒŒæ™¯å˜›ï¼Œæœ€åæ‹¼å›¾å˜›ã€‚å½“ç„¶æ‹¼å›¾ä¸ä¸€å®šè¦åœ¨illustratoré‡Œæ‹¼ï¼Œæ¯”å¦‚ä½ ç”¨gridï¼Œå…ˆç”»å¥½åæ ‡è½´ï¼Œç„¶åæ°´å¹³ä¸Šå®šä¹‰å‡ ä¸ªä¸€æ ·å¤§çš„viewportï¼Œæ¯ä¸€ä¸ªviolinéƒ½ç”»åœ¨ç›¸åº”çš„viewporté‡Œé¢ï¼Œå¯¹äºç”»å›¾å‡½æ•°æ¥è¯´ï¼Œviewportå°±æ˜¯æ•´ä¸ªç”»å¸ƒäº†ï¼ˆè™½ç„¶åªæ˜¯ç”»å¸ƒé‡Œçš„ä¸€å—åŒºåŸŸï¼‰ï¼Œæ‰€ä»¥ä½ è¦ä¹ˆç”»violinï¼Œè¦ä¹ˆåœ¨ç”»ä¹‹å‰å…ˆç”»ä¸ªçŸ©é˜µï¼Œä¸€è·¯ç”»ä¸‹æ¥ï¼Œä»£ç å¯ä»¥ç›´æ¥ç”Ÿæˆè¿™æ ·çš„å›¾ï¼Œä½†è¿™å›¾æ¯ä¸€ä¸ªviolinéƒ½æ˜¯ç‹¬ç«‹ç”»çš„ï¼ˆå½“ç„¶ä¹Ÿä¸æ˜¯å®Œå…¨ç‹¬ç«‹ï¼Œæ¯ä¸€ä¸ªæ°´å¹³ä¸Šçš„ylimæ˜¯æœ‰ç»Ÿä¸€çš„ï¼‰ï¼Œå°±ç®—ä»£ç ä¸€æ­¥ç”Ÿæˆï¼Œä¹Ÿè·Ÿillustratoré‡Œæ‹¼æ²¡ä¸¤æ ·ã€‚</p>

<hr />

<p>è¿™é‡Œæˆ‘è¦æ•™ä½ ç”¨ggplot2è‡ªåŠ¨ç”Ÿæˆï¼Œå…¶å®è§£å†³æ€è·¯æ—©å·²æ¨é€è¿‡ï¼Œè¯·çœ‹ã€Š<a href="http://mp.weixin.qq.com/s/PnbasfW4HKILuZNdrLVX_g">facet_plotï¼šåŠ å›¾å±‚åˆ°ç‰¹å®šåˆ†é¢ï¼Œæ–¹æ³•äºŒ</a>ã€‹ï¼Œä¹Ÿæ­£å¦‚æˆ‘åœ¨ã€Š<a href="http://mp.weixin.qq.com/s/8bzkJY1zzNc83iBUq4Fy1w">ä»€ä¹ˆï¼ä½ çš„å›¾ä¸Šæœ‰ä¸€åŒçœ‹ä¸è§çš„æ‰‹</a>ã€‹é‡Œè¯´çš„ï¼Œä½ ä»¬ä»¥ä¸ºæˆ‘åœ¨æ•™ggtreeï¼Œå…¶å®åŒæ—¶åœ¨æ•™ggplot2ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
      </div>
    </div>
    <div class="modal-footer">
      <p class="results-count text-medium"
         data-message-zero="no post found"
         data-message-one="1 post found"
         data-message-other="{n} posts found">
         181 posts found
      </p>
    </div>
  </div>
</div>
    
  
    
    <div id="cover" style="background-image:url('https://guangchuangyu.github.io/cn/images/cover.jpg');"></div>
  


    
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/highlight.min.js"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.js"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.js"></script>


<script src="https://guangchuangyu.github.io/cn/js/script-wl33z0n6ocaypepiqrazthtivfrliqijej4rq8ek8gvrv1awftmgjuv8k4zc.min.js"></script>

<script>
$(document).ready(function() {
  hljs.configure({ classPrefix: '', useBR: false });
  $('pre.code-highlight').each(function(i, block) {
    var code = "";
    hljs.highlightAuto(block.innerText).value.split(/\r\n|\r|\n/).forEach(function(line) {
      code += "<span class=\"line\">" + line + "</span><br>";
    });
    if (code.length > 0) {
      block.innerHTML = code;  
    }
  });
  $('pre > code').each(function(i, block) {
    $(this).addClass('codeblock');
    hljs.highlightBlock(block);
  });
});
</script>

  
    
      <script>
        var disqus_config = function () {
          this.page.url = 'https:\/\/guangchuangyu.github.io\/cn\/2011\/07\/a-simple-gene-finder\/';
          
            this.page.identifier = '\/2011\/07\/a-simple-gene-finder\/'
          
        };
        (function() {
          
          
          if (window.location.hostname == "localhost") {
            return;
          }
          var d = document, s = d.createElement('script');
          var disqus_shortname = 'gcyu';
          s.src = '//' + disqus_shortname + '.disqus.com/embed.js';

          s.setAttribute('data-timestamp', +new Date());
          (d.head || d.body).appendChild(s);
        })();
      </script>
    
  





    
  </body>
</html>

