

  
    
  


  




  


  

<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="Hugo 0.48 with theme Tranquilpeak 0.4.3-BETA">
    <title>a simple gene finder</title>
    <meta name="author" content="Guangchuang Yu">
    <meta name="keywords" content="">

    <link rel="icon" href="https://guangchuangyu.github.io/cn/favicon.png">
    

    
    <meta name="description" content="上了几天的课，http://ibw2011.fmmu.edu.cn/schedule.htm
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。

这个纯属练习用，没啥实用价值。


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    <meta property="og:description" content="上了几天的课，http://ibw2011.fmmu.edu.cn/schedule.htm
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。

这个纯属练习用，没啥实用价值。


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    <meta property="og:type" content="blog">
    <meta property="og:title" content="a simple gene finder">
    <meta property="og:url" content="/2011/07/a-simple-gene-finder/">
    <meta property="og:site_name" content="Guangchuang Yu">
    <meta name="twitter:card" content="summary">
    <meta name="twitter:title" content="Guangchuang Yu">
    <meta name="twitter:description" content="上了几天的课，http://ibw2011.fmmu.edu.cn/schedule.htm
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。

这个纯属练习用，没啥实用价值。


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    
      <meta name="twitter:creator" content="@guangchuangyu">
    
    

    
    

    
      <meta property="og:image" content="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">
    

    
    
    

    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.css" integrity="sha256-vuXZ9LGmmwtjqFX1F+EKin1ThZMub58gKULUyf0qECk=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" integrity="sha256-SEa4XYAHihTcEP1f5gARTB2K26Uk8PsndQYHQC1f4jU=" crossorigin="anonymous" />
    
    
    <link rel="stylesheet" href="https://guangchuangyu.github.io/cn/css/style-jsjn0006wyhpyzivf6yceb31gvpjatbcs3qzjvlumobfnugccvobqwxnnaj8.min.css" />
    
    

    
      
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-77901140-1', 'auto');
	
	ga('send', 'pageview');
}
</script>

    
    
  </head>

  <body>
    <div id="blog">
      <header id="header" data-behavior="4">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="https://guangchuangyu.github.io/cn/">Guangchuang Yu</a>
  </div>
  
    
      <a class="header-right-picture "
         href="https://guangchuangyu.github.io/cn/#about">
    
    
    
      
        <img class="header-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=90" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="4">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <a href="https://guangchuangyu.github.io/cn/#about">
          <img class="sidebar-profile-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Guangchuang Yu</h4>
        
          <h5 class="sidebar-profile-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/categories">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/tags">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/archives">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://github.com/GuangchuangYu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-github"></i>
      
      <span class="sidebar-button-desc">GitHub</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/index.xml">
    
      <i class="sidebar-button-icon fa fa-lg fa-rss"></i>
      
      <span class="sidebar-button-desc">RSS</span>
    </a>
  </li>


    </ul>
  </div>
</nav>

      

      <div id="main" data-behavior="4"
        class="
               hasCoverMetaIn
               ">
        <article class="post" itemscope itemType="http://schema.org/BlogPosting">
          
          
            <div class="post-header main-content-wrap text-left">
  
    <h1 class="post-title" itemprop="headline">
      a simple gene finder
    </h1>
  
  
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2011-07-10T09:04:46&#43;08:00">
        
  July 10, 2011

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/cn/categories/conference">conference</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/cn/categories/r">R</a>
    
  

  </div>

</div>
          
          <div class="post-content markdown" itemprop="articleBody">
            <div class="main-content-wrap">
              <p>上了几天的课，<a href="http://ibw2011.fmmu.edu.cn/schedule.htm">http://ibw2011.fmmu.edu.cn/schedule.htm</a>
今天就上完了，只完成了project 1，想写gibbs sampling，但是没搞明白，汗。</p>

<p>这个纯属练习用，没啥实用价值。</p>

<blockquote>
<p>Course Projects:</p>

<p>Project 1: Implementation of a simple gene finder</p>

<p>GOAL</p>

<p>Build a simple codon-usage based gene finder for finding genes in
E.coli.</p>

<p>Procedure</p>

<p>Collect 100 gene sequences from the bacterium E. coli in the genbank
(<a href="http://www.ncbi.nlm.nihh.gov">http://www.ncbi.nlm.nihh.gov</a>). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:</p>

<p>Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder –i inputfile</p>

<p>Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.</p>

<p>The output should be printed to the standard output as (xxx stands for
the likelihood)</p>

<p>ORF1: xxx ORF2: xxx</p>
</blockquote>

<p></p>

<pre><code class="language-r">
###########################################################
##                                                       ##
##  Dragonstar Course Project (IBW 2011)                 ##
##                                                       ##
##  Project 1: Implementation of a simple gene finder    ##
##  Author Guangchuang your                              ##
##  Xi'an (2011-07-08)                                   ##
##                                                       ##
###########################################################



##' .. content for \description{} (ECgnfinder: An R program to predict prokaryotic genes (e.g. E.coli) using 1st Markov Chain) ..
##'
##' .. content for \details{} ..
##' @title ECgnfinder
##' @param traningSequence : A FASTA file containing multiple sequences for training set
##' @param testingSequence : A FASTA file containing multiple sequences for testing
##' @param backgroundSequence : A FASTA file containing multiple sequences for building the background model
##' @return log likelihood ratio
##' @author Guangchuang Yu
ECgnfinder &lt;- function(trainingSequence, testingSequence, backgroundSequence, print=FALSE) {
    print (&quot;reading training sequence file...&quot;)
    trainingFasta &lt;- readFASTA(trainingSequence)
    trainSeq &lt;- sapply(trainingFasta, function(i) i$seq)

    print(&quot;building Transition Probability Matrix...&quot;)
    TPM &lt;- getTPM(trainSeq)

    print(&quot;building codon usage table...&quot;)
    codonUsage &lt;- getCodonUsage(trainSeq)

    print(&quot;building NucleoTide table...&quot;)
    NtUsage &lt;- getNtUsage(backgroundSequence)


    print(&quot;reading testing sequence file...&quot;)
    testing &lt;- readFASTA(testingSequence)
    testingSeq &lt;- sapply(testing, function(i) i$seq)

    x &lt;- lapply(testingSeq, logLikelihood, transition=TPM, codonusage=codonUsage, background=NtUsage)
    names(x) &lt;- paste(&quot;Sequence&quot;, seq_along(testingSeq), sep=&quot;&quot;)
    if (print) {
        x
    }
    return(x)
}


##' .. content for \description{} (read FASTA file) ..
##'
##' .. content for \details{} ..
##' @title readFASTA
##' @param file: A FASTA file
##' @return a list of list which containing desc and seq.
##' @author Guangchuang Yu
readFASTA &lt;- function(file) {
    x=scan(file, what=&quot;&quot;, sep=&quot;\n&quot;)
    idx &lt;- grep(&quot;&gt;&quot;, x)
    fasta &lt;- list()
    for (i in 1:length(idx)) {
            desc &lt;- x[idx[i]]
            if (i == length(idx)) {
                stop &lt;- length(x)
            } else {
                stop &lt;- idx[i+1]-1
            }
            seq.idx &lt;- seq(idx[i]+1, stop)
            s &lt;- x[seq.idx]
            s &lt;- paste(s, collapse=&quot;&quot;)
            ss &lt;- list(desc=desc, seq=s)
            fasta[i] &lt;- list(ss)
    }
    return(fasta)
    }


##' Build a transition probability matrix from the given Sequence set.
##'
##' @title getTPM
##' @param sequences : A vector of Sequences.
##' @return transition probability matrix
##' @author Guangchuang Yu
getTPM &lt;- function(sequences) {
    codonLength &lt;- 3
    codons &lt;- getCodon()
    
    ## calculate a transition count matrix
    transition &lt;- matrix(1, nrow=length(codons), ncol=length(codons))  ## pseudo count
    colnames(transition) &lt;- codons
    rownames(transition) &lt;- codons

    for (x in 1:length(sequences)) {
        s &lt;- sequences[x]
        len &lt;- nchar(s)
        ## drop the last alphabets which is too short (

        idx &lt;- seq(1, len, codonLength)
        for (i in seq_along(idx)) {
            ## codonI--codonJ
            codonI &lt;- substr(s,idx[i], idx[i]+codonLength-1)
            if (i != length(idx)) {
                codonJ &lt;- substr(s,idx[i+1], idx[i+1]+codonLength-1)
            }
            if (codonI %in% codons &amp; codonJ %in% codons) {
                transition[codonI, codonJ] &lt;- transition[codonI, codonJ] +1
            } else {
                warning(&quot;codon &quot;, codonI, &quot; or &quot;, codonJ, &quot; which does not defined was detected...&quot;)
            }
        }
    }

    ## Calculate Transition Probability Matrix..
    rowsum &lt;- rowSums(transition)
    TPM &lt;- transition/rowsum
    return(TPM)
}

getCodon &lt;- function() {
    codons &lt;- c(&quot;TTT&quot;, &quot;TTC&quot;, &quot;TTA&quot;, &quot;TTG&quot;,
                &quot;CTT&quot;, &quot;CTC&quot;, &quot;CTA&quot;, &quot;CTG&quot;,
                &quot;ATT&quot;, &quot;ATC&quot;, &quot;ATA&quot;, &quot;ATG&quot;,
                &quot;GTT&quot;, &quot;GTC&quot;, &quot;GTA&quot;, &quot;GTG&quot;,
                &quot;TCT&quot;, &quot;TCC&quot;, &quot;TCA&quot;, &quot;TCG&quot;,
                &quot;CCT&quot;, &quot;CCC&quot;, &quot;CCA&quot;, &quot;CCG&quot;,
                &quot;ACT&quot;, &quot;ACC&quot;, &quot;ACA&quot;, &quot;ACG&quot;,
                &quot;GCT&quot;, &quot;GCC&quot;, &quot;GCA&quot;, &quot;GCG&quot;,
                &quot;TAT&quot;, &quot;TAC&quot;, &quot;TAA&quot;, &quot;TAG&quot;,
                &quot;CAT&quot;, &quot;CAC&quot;, &quot;CAA&quot;, &quot;CAG&quot;,
                &quot;AAT&quot;, &quot;AAC&quot;, &quot;AAA&quot;, &quot;AAG&quot;,
                &quot;GAT&quot;, &quot;GAC&quot;, &quot;GAA&quot;, &quot;GAG&quot;,
                &quot;TGT&quot;, &quot;TGC&quot;, &quot;TGA&quot;, &quot;TGG&quot;,
                &quot;CGT&quot;, &quot;CGC&quot;, &quot;CGA&quot;, &quot;CGG&quot;,
                &quot;AGT&quot;, &quot;AGC&quot;, &quot;AGA&quot;, &quot;AGG&quot;,
                &quot;GGT&quot;, &quot;GGC&quot;, &quot;GGA&quot;, &quot;GGG&quot;
                )
    return(codons)
}


##' .. content for \description{} (estimate codon usage table based on training sequences) ..
##'
##' .. content for \details{} ..
##' @title getCodonUsage
##' @param sequences : A vector of sequence set.
##' @return A data.frame containing two columns of codons and Freq (frequency probabilities)
##' @author Guangchuang Yu
getCodonUsage &lt;- function(sequences) {
    codons &lt;- getCodon()
    codon.table &lt;- data.frame(codons=codons, Freq=rep(0, length(codons)))
    rownames(codon.table) = codons

    for (i in 1:length(sequences)) {
        codon &lt;- codonCount(sequences[i])
        rowname.idx &lt;- as.character(codon[,1])
        codon.table[rowname.idx,2] &lt;- codon.table[rowname.idx,2] + codon[,2]
    }
    codon.table[,2] &lt;- codon.table[,2]/sum(codon.table[,2])
                                        #print(codon.table)  #### print result.
    return(codon.table)
}

##' .. content for \description{} (no empty lines) ..
##'
##' .. content for \details{} ..
##' @title codonCount
##' @param seqStr : A sequence string
##' @return A data.frame containing two columns of codons and their counts
##' @author Guangchuang Yu
codonCount &lt;- function(seqStr) {
    codonLength=3
    len &lt;- nchar(seqStr)
    ## drop the last alphabets which is too short (

    idx &lt;- seq(1, len,codonLength)
    codons &lt;- c()
    for ( i in 1:length(idx)) {
        codon &lt;- substr(seqStr, idx[i], idx[i]+codonLength-1)
        codons &lt;- c(codons, codon)
    }
    codons.df &lt;- as.data.frame(table(codons))
    return(codons.df)
}

##' .. content for \description{} (reverse complementary sequence) ..
##'
##' .. content for \details{} ..
##' @title revcom
##' @param seqStr : A sequence string
##' @return A reverse complementary sequence
##' @author Guangchuang Yu
revcom &lt;- function(seqStr) {
    seqStr.str &lt;- unlist(strsplit(seqStr, split=&quot;&quot;))
    idxT &lt;- which(seqStr.str==&quot;T&quot;)
    idxG &lt;- which(seqStr.str==&quot;G&quot;)
    idxA &lt;- which(seqStr.str==&quot;A&quot;)
    idxC &lt;- which(seqStr.str==&quot;C&quot;)
    seqStr.str[idxT] &lt;- &quot;A&quot;
    seqStr.str[idxA] &lt;- &quot;T&quot;
    seqStr.str[idxG] &lt;- &quot;C&quot;
    seqStr.str[idxC] &lt;- &quot;G&quot;
    seqStr.rc &lt;- rev(seqStr.str)
    seqStr.rc &lt;- paste(seqStr.rc, collapse=&quot;&quot;)
    return(seqStr.rc)
}

##' .. content for \description{} (calculate NT frequency probabilities of a given sequence set) ..
##'
##' .. content for \details{use to estimate the background/random NT usage} ..
##' @title getNtUsage
##' @param file : A FASTA file
##' @return A data.frame containing NTs and their corresponding frequency probabilities.
##' @author Guangchuang Yu
getNtUsage &lt;- function(file) {
    print(&quot;reading FASTA file&quot;)
    fasta &lt;- readFASTA(file)
    sequences &lt;- sapply(fasta, function(i) i$seq)
    x &lt;- sapply(sequences, strsplit, split=&quot;&quot;)
    x &lt;- unlist(x)
    NTprob &lt;- as.data.frame(table(x))
    NTprob[,2] &lt;- NTprob[,2]/sum(NTprob[,2])
    rownames(NTprob) &lt;- NTprob[,1]
    return(NTprob)
}

##' .. content for \description{} (calculate the log likelihood of the six possible reading frame of a given sequence) ..
##'
##' .. content for \details{} ..
##' @title logLikelihood
##' @param sequence : A sequence string
##' @param transition : Transition Probatility Matrix
##' @param codonusage : Codon Usage data.frame
##' @param background : NT Usage data.frame for background/random model
##' @return Log likelihood
##' @author Guangchuang Yu
logLikelihood &lt;- function(sequence, transition, codonusage, background) {
    codonLength &lt;- 3
    orf1 &lt;- sequence
    orf2 &lt;- substr(sequence, 2, nchar(sequence))
    orf3 &lt;- substr(sequence, 3, nchar(sequence))
    seq.rc &lt;- revcom(sequence)
    orf4 &lt;- seq.rc
    orf5 &lt;- substr(seq.rc, 2, nchar(seq.rc))
    orf6 &lt;- substr(seq.rc, 3, nchar(seq.rc))
    ORF &lt;- c(orf1, orf2, orf3, orf4, orf5, orf6)
    names(ORF) &lt;- c(&quot;ORF1&quot;, &quot;ORF2&quot;, &quot;ORF3&quot;, &quot;ORF4&quot;, &quot;ORF5&quot;, &quot;ORF6&quot;)

    result &lt;- data.frame(ORF=names(ORF), logLikelihood=rep(0,6))
    for (j in seq_along(ORF)) {
        firstCodon &lt;- substr(ORF[j], 1, 3)
        if (firstCodon %in% codonusage[,1]) {
            initProb &lt;- log(codonusage[firstCodon,2])
        } else {
            initProb &lt;- log(1/64)
        }

        transitionCount &lt;- matrix(0, nrow=nrow(transition), ncol=ncol(transition))
        rownames(transitionCount) &lt;- rownames(transition)
        colnames(transitionCount) &lt;- colnames(transition)

        codons &lt;- rownames(transition)

        len &lt;- nchar(ORF[j])
        len &lt;- len - len %% codonLength - codonLength

        idx &lt;- seq(1, len, codonLength)
        for (i in seq_along(idx)) {
            ## codonI--codonJ
            codonI &lt;- substr(ORF[j],idx[i], idx[i]+codonLength-1)
            if (i != length(idx)) {
                codonJ &lt;- substr(ORF[j],idx[i+1], idx[i+1]+codonLength-1)
            }
            if (codonI %in% codons &amp; codonJ %in% codons) {
                transitionCount[codonI, codonJ] &lt;- transitionCount[codonI, codonJ] +1
            } else {
                warning(&quot;codon &quot;, codonI, &quot; or &quot;, codonJ, &quot; which does not defined was detected...&quot;)
            }
        }

        postTransition &lt;- log(transition) * transitionCount
        posterior &lt;- initProb + sum(postTransition)

        NTseq &lt;- unlist(strsplit(ORF[j], split=&quot;&quot;))
        orf.len &lt;- nchar(ORF[j]) - nchar(ORF[j]) %% codonLength
        NTseq &lt;- NTseq[1:orf.len]
        randomModel &lt;- sum(log(background[NTseq,2]))

        likelihood &lt;- posterior - randomModel
        result[j, 2] &lt;- likelihood
    }
    return(result)
}
</code></pre>
              
            </div>
          </div>
          <div id="post-footer" class="post-footer main-content-wrap">
            
              
                
                
                  <div class="post-footer-tags">
                    <span class="text-color-light text-small">TAGGED IN</span><br/>
                    
  <a class="tag tag--primary tag--small" href="https://guangchuangyu.github.io/cn/tags/ibw/">IBW</a>

                  </div>
                
              
            
            
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
          
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/07/ibw2011/" data-tooltip="IBW2011">
              
                  <i class="fa fa-angle-left"></i>
                  <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
                </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/06/screen-shot/" data-tooltip="screen shot">
              
                  <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                  <i class="fa fa-angle-right"></i>
                </a>
            </li>
          
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


            
              
                <div id="disqus_thread">
  <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
              
            
          </div>
        </article>
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2018 Guangchuang Yu. All Rights Reserved
  </span>
</footer>

      </div>
      <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
        
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
          
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/07/ibw2011/" data-tooltip="IBW2011">
              
                  <i class="fa fa-angle-left"></i>
                  <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
                </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/06/screen-shot/" data-tooltip="screen shot">
              
                  <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                  <i class="fa fa-angle-right"></i>
                </a>
            </li>
          
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


      </div>
      <div id="share-options-bar" class="share-options-bar" data-behavior="4">
  <i id="btn-close-shareoptions" class="fa fa-close"></i>
  <ul class="share-options">
    
  </ul>
</div>
<div id="share-options-mask" class="share-options-mask"></div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Guangchuang Yu</h4>
    
      <div id="about-card-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Postdoc researcher
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        Hong Kong
      </div>
    
  </div>
</div>

    <div id="algolia-search-modal" class="modal-container">
  <div class="modal">
    <div class="modal-header">
      <span class="close-button"><i class="fa fa-close"></i></span>
      <a href="https://algolia.com" target="_blank" rel="noopener" class="searchby-algolia text-color-light link-unstyled">
        <span class="searchby-algolia-text text-color-light text-small">by</span>
        <img class="searchby-algolia-logo" src="https://www.algolia.com/static_assets/images/press/downloads/algolia-light.svg">
      </a>
      <i class="search-icon fa fa-search"></i>
      <form id="algolia-search-form">
        <input type="text" id="algolia-search-input" name="search"
          class="form-control input--large search-input" placeholder="Search" />
      </form>
    </div>
    <div class="modal-body">
      <div class="no-result text-color-light text-center">no post found</div>
      <div class="results">
        
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/09/install-bioconductor-package/">
                <h3 class="media-heading">不用biocLite安装Bioconductor包</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>Bioconductor包的标准安装方法是：</p>

<pre><code>source(&quot;https://bioconductor.org/biocLite.R&quot;)
biocLite(pkg_name)
</code></pre>

<p>这其实是非常不好的，多年以来，一直没有改进，试想一下，bioconductor的网站如果被人黑了，放一段危险的代码，然后你却傻傻地去<code>source</code>它，真的是画面太美！</p>

<p>Bioconductor终于是想要改变了，往CRAN上扔了一个叫<code>BiocManager</code>的包，以后通过CRAN安装这个包，再通过这个包来安装Bioconductor的包，显得安全多了，也不用大家费辣么多口水说你得先<code>source</code>一段网上的代码，当然这个<code>BiocManager</code>还不成熟，起码还没有在<code>BiocInstaller</code>加载的时候提示大家转而用<code>BiocManager</code>，也没有在Bioconductor的网页上介绍使用它来安装。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/09/facet_labeller/">
                <h3 class="media-heading">用facet_labeller改facet_plot的panel名</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><blockquote>
<p>I post this because I am facing the same problem: I want to change the text of the label (not the font, the face, the colour, etc: the text itself) of the leftmost panel which is automatically called &ldquo;Tree&rdquo; by ggtree. As far as I understand, and contrary to what is posted below, the &ldquo;theme&rdquo; thing of ggplot2 only allows to change the appearance of the panel header, not the text of the panel title itself&hellip; So how to go about that?</p>
</blockquote>

<p>这是<a href="https://groups.google.com/d/msgid/bioc-ggtree/154a6acc-bf78-40d9-ad95-8dc1877a4f97%40googlegroups.com?utm_medium=email&amp;utm_source=footer">google group</a>里的问题，说的是<code>facet_plot</code>会把画树的panel叫做Tree，而他想改名，于是我就写了一个函数，<code>facet_labeller</code>来回应这个问题。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/08/ggplot-3d/">
                <h3 class="media-heading">用ggplot2画3D</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Aug 8, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>在《<a href="http://mp.weixin.qq.com/s/BPY4dPFrrNrDKccMvax07w">扪心自问，meme几何？</a>》一文中，我画了一只兔子，今天给大家演示一下画一只3D版本的兔子，我们知道<code>ggplot2</code>不支持3D，只能画2D的图，但3D其实只是视觉上的，显示在你屏幕上的，同样只是2D。所以我们要做的，就是把3D的坐标转换成2D，然后画出2D的图片来，而这2D看上去有3D的效果。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/08/arch-downgrade/">
                <h3 class="media-heading">Arch之软件降级</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Aug 8, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>用Arch Linux就是每天睡觉关机前滚一下，然而有时候有些软件升级滚出问题，比如<code>glibc-2.28</code>就导致了<code>R</code>安装包有问题，会抛出下面的错误：</p>

<pre><code>terminate called after throwing an instance of 'std::runtime_error'
  what():  Mutex creation failed
</code></pre>

<p>用<code>blogdown</code>编译博客也同样会抛出这个错误。</p>

<p>然后<code>electron</code>也有问题，《<a href="http://mp.weixin.qq.com/s/L556YNtVSKf84cEFdEZVBQ">开源的macOS和linux版微信</a>》跑着跑着突然CPU就狂飙。</p>

<p>解决办法就是安装回老版本，但<code>pacman</code>永远是最新的，当然可以去git/svn仓库里找老版本的<code>PKGBUILD</code>，然后再<code>makepkg</code>，但经常会在<code>signature checking</code>这步报错，你还得到导入<code>keyring</code>，挺麻烦。</p>

<p>这时候我找到了<code>downgrade</code>脚本:</p>

<blockquote>
<p>Bash script for downgrading one or more packages to a version in your cache or the A.L.A.</p>
</blockquote>

<p>AUR有收录，安装非常简单：</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/08/pheatmap-class/">
                <h3 class="media-heading">pheatmap类属性：一个更好用的pheatmap</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Aug 8, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p><code>pheatmap</code>我其实用的次数不多，但在帮别人解决一些问题之后，比如：</p>

<ul>
<li><a href="http://mp.weixin.qq.com/s/Xhi_ZkrEzUVdnCu7ECBS4w">听说你还不会画热图</a></li>
<li><a href="https://mp.weixin.qq.com/s/yVLgGQQi0CQ9Q9UfIKl5aA">不同数据集画出的热图，用同样的颜色区间上色?</a></li>
</ul>

<p>我发现这包的交互能力是比较差的，它就像传统的<code>plot</code>函数一样，命令一打下去，就直接画图，虽然它其实是有输出的，但事实上超出了一般用户的理解能力，它的输出，比如说可以用<code>cowplot</code>来拼图，我在上面链接的那篇文章中皆来演示，然而很多人问我，怎么把<code>pheatmap</code>画的热图和<code>ggplot2</code>的图，用<code>cowplot</code>来拼。这就是因为拼图的方法超出了很多的理解范围，看过演示，也只能照猫画老虎，过后也就忘记了。有没有办法用大家可以理解的方法来拼？这样就不太容易忘记，答案是肯定的，我下次再讲。今天先讲一讲其前提条件。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/linux_qq/">
                <h3 class="media-heading">酗酒的Linux不是好企鹅</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>国内的办公环境，离不开QQ等Windows软件，以前我通过《<a href="http://mp.weixin.qq.com/s/7AdkrX3hZYdKNP4YIhgZLA">安装黑苹果</a>》来解决，如果要用Linux呢？我们不禁想问，<a href="https://solarianprogrammer.com/2017/04/16/windows-susbsystem-for-linux-xfce-4/">连Windows都能跑Linux程序了</a>，Linux到底什么时候能跑Windows程序？</p>

<h2 id="干了这杯酒-不干没鸡鸡">干了这杯酒，不干没鸡鸡！</h2>

<p><img src="https://lh3.googleusercontent.com/-dfDJJYSQ4MM/V-PSb7LmfPI/AAAAAAAAEy8/EXNlBjf7GZI_aAfC-lwQx2bHkppC5ereQCJoC/w530-h530-n-rw/e446d718-5ca4-42ec-8d59-45f4630b211d" alt="" /></p>

<p>Wine虽然是一个解决方案，而且现在也有现成的方案，比如《<a href="http://mp.weixin.qq.com/s/4yWzHxjB9WL33ujagi5BAw">你们想要的：QQ跑在Linux上</a> 》，再比如有打包好的深度QQ等。但我不想在我64位的系统里装一堆32位的兼容包，也不想（多多少少得）折腾Wine。</p>

<h2 id="无需安装-开箱即用">无需安装，开箱即用？</h2>

<p>这显然是最好的方案，比如在苹果系统上，我已经打包了《<a href="http://mp.weixin.qq.com/s/F0TxMk1CrYVNvxWqZgv-Aw">OSX版BioEdit</a>》和《<a href="https://mp.weixin.qq.com/s/P3WhPkTrf2dUkzvOba3hlQ">独家：SciHub苹果桌面版</a>》，你只需要拖到<code>Applications</code>里，双击就能用。</p>

<p>这有什么好处？第一你不用去系统里装Wine以及一堆依赖包，第二你不需要去配置Wine，第三这个运行环境是与你的系统分离的（系统是干净的）。对于用户来说，傻瓜化开箱即用，对于系统来说，干净不影响。这是最好的方案。谁愿意去装一堆没用的兼容<code>libs</code>，谁愿意去折腾老半天还用不上！当然有一点点不好的地方，是打包的软件体积稍大一点，因为你要打包整个运行环境嘛，但现在最不值钱的就是存储空间了。</p>

<p>所以最好的方案是像Mac这样，可以把运行环境分离出来，系统干净不折腾！当然再者还得有像我这样的好心人去打包。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/partial-match-when-subsetting-rowname-of-data-frame/">
                <h3 class="media-heading">R总给我惊喜</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><blockquote>
<p>Hello!
First of all, I would like to thank you for this wonderful and very powerful package!</p>

<p>I have tried to plot a phylogenetic tree with heatmap of associated matrix (with <code>gheatmap</code>).
I found that the row names of matrix doesn&rsquo;t exactly match to the tip names of tree in case if there is missing data in associated matrix.
In other words, if we have two species (e.g., Species<strong>198</strong> and Species<strong>1981</strong>), but only one of them is represented in the associated matrix, we will have colored cells for both species in the heatmap.</p>
</blockquote>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/clusterprofiler-ath/">
                <h3 class="media-heading">拟南芥的富集分析</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>「知识星球」里的问题，<a href="https://t.zsxq.com/aEyneA2">https://t.zsxq.com/aEyneA2</a>，拟南芥的ID怎么搞，这个内置是支持的。</p>

<p>比如我随便搞几个ID来演示一下：</p>

<pre><code>&gt; require(clusterProfiler)
Loading required package: clusterProfiler

clusterProfiler v3.9.0  For help: https://guangchuangyu.github.io/software/clusterProfiler

If you use clusterProfiler in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for 
comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.

g &lt;- c(&quot;AT5G61600&quot;, &quot;AT2G44230&quot;, &quot;AT1G49130&quot;, &quot;AT1G30310&quot;, &quot;AT3G18560&quot;,
&quot;AT2G15960&quot;, &quot;AT2G29290&quot;, &quot;AT1G08930&quot;, &quot;AT2G44460&quot;, &quot;AT3G46600&quot;,
&quot;AT5G24380&quot;, &quot;AT3G59270&quot;, &quot;AT5G55780&quot;, &quot;AT2G45090&quot;, &quot;AT5G38430&quot;,
&quot;AT2G17040&quot;, &quot;AT4G35770&quot;, &quot;AT5G30341&quot;, &quot;AT1G56320&quot;, &quot;AT2G16367&quot;)
</code></pre>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/scholar-impact-factor/">
                <h3 class="media-heading">scholar包检索期刊影响因子（2018年最新版本）</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>在《<a href="https://mp.weixin.qq.com/s/kBnLyNl2anXuMa6sDepRQw">一个要复活的R包和一个404的网站</a>》一文中第一次介绍了scholar包，它可以应用于《<a href="https://mp.weixin.qq.com/s/IBrEOh92r_90DRbGb3XlKg">自动抓取论文的引用数据</a>》，虽然数据源来自于Google Scholar，国内用户可能访问不了，但今天要介绍的这个功能，所有人都是可以用的，它允许我们检索期刊的影响因子，在《<a href="https://mp.weixin.qq.com/s/qnuiepL4-nSiSWJVJH54ug">scholar包支持检索期刊影响因子</a>》一文中第一次介绍了这个功能，当时是2016年的影响因子（2017年发布），现在数据已经更新至2018年6月26号发布的2017年影响因子。</p>

<pre><code>&gt; library(scholar)
&gt; jn = c(&quot;bioinformatics&quot;, &quot;methods in ecology and evolution&quot;,
+     &quot;molecular biosystems&quot;, &quot;molecular biology and evolution&quot;)
&gt; get_impactfactor(jn)
# A tibble: 4 x 4
  Journal                          Cites ImpactFactor Eigenfactor
  &lt;chr&gt;                            &lt;dbl&gt;        &lt;dbl&gt;       &lt;dbl&gt;
1 BIOINFORMATICS                   95300         5.48      0.201
2 Methods in Ecology and Evolution  8866         6.36      0.0370
3 Molecular BioSystems              7269         2.76      0.0175
4 MOLECULAR BIOLOGY AND EVOLUTION  44664        10.2       0.102
</code></pre>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/06/geom_worldcup2018/">
                <h3 class="media-heading">2018世界杯专属图层</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jun 6, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>说到世界杯，我是不追的，但1998年法国世界杯我还是有印象的，那时候读初中的我，一直在集邮，那年代没有网购，通讯基本靠吼，当年的我已经是个老司机了，通过邮政汇款，买了好多东西，包括1998年世界杯从法国寄往中国的首日封，当然锁在老家的柜子里，尘封多年，所以我现在也没办法随手拍一张照片给大家分享。</p>

<p>另一件事，同样是1998年，那年升高一，开学第一天，老师让大家自我介绍一下，有位帅哥在见面第一天，就立了flag，说自己中考没考好，是因为看世界杯了，他高考必定要怎么样怎么样，毕竟世界杯是四年一次。你一个交钱进学校的学渣，牛逼吹大了，我印象深刻，所以我后来留意了一下这家伙高考上了那家高校，结果是「中山大学网络教育学院」，也就是电视大学！真的是情节反转如疾风！</p>

<p>《<a href="https://mp.weixin.qq.com/s/EojwBbg0MUWijwGeNIjntg">转发5个群，Y叔送你一顶帽子</a>》这一篇文章写的是加圣诞帽子，但你要加个C罗、梅西，有什么差别吗？没有！你要加某球队的队服，有什么差别吗？没有。所以要制作世界杯专属头像，已然可以用ggimage纯代码生成。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
      </div>
    </div>
    <div class="modal-footer">
      <p class="results-count text-medium"
         data-message-zero="no post found"
         data-message-one="1 post found"
         data-message-other="{n} posts found">
         232 posts found
      </p>
    </div>
  </div>
</div>
    
  
    
    <div id="cover" style="background-image:url('https://guangchuangyu.github.io/cn/images/cover.jpg');"></div>
  


    
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.4/jquery.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.7/js/jquery.fancybox.min.js" integrity="sha256-GEAnjcTqVP+vBp3SSc8bEDQqvWAZMiHyUSIorrWwH50=" crossorigin="anonymous"></script>


<script src="https://guangchuangyu.github.io/cn/js/script-qi9wbxp2ya2j6p7wx1i6tgavftewndznf4v0hy2gvivk1rxgc3lm7njqb6bz.min.js"></script>




  
    
      <script>
        var disqus_config = function () {
          this.page.url = 'https:\/\/guangchuangyu.github.io\/cn\/2011\/07\/a-simple-gene-finder\/';
          
            this.page.identifier = '\/2011\/07\/a-simple-gene-finder\/'
          
        };
        (function() {
          
          
          if (window.location.hostname == "localhost") {
            return;
          }
          var d = document, s = d.createElement('script');
          var disqus_shortname = 'gcyu';
          s.src = '//' + disqus_shortname + '.disqus.com/embed.js';

          s.setAttribute('data-timestamp', +new Date());
          (d.head || d.body).appendChild(s);
        })();
      </script>
    
  



    
  </body>
</html>

