

  
    
  


  




  


  

<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="Hugo 0.30.2">
    <meta name="theme" content="Tranquilpeak 0.3.1-BETA">
    <title>a simple gene finder</title>
    <meta name="author" content="Guangchuang Yu">
    <meta name="keywords" content="">

    <link rel="icon" href="https://guangchuangyu.github.io/cn/favicon.png">
    

    
    <meta name="description" content="ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œhttp://ibw2011.fmmu.edu.cn/schedule.htm
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚

è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    <meta property="og:description" content="ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œhttp://ibw2011.fmmu.edu.cn/schedule.htm
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚

è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    <meta property="og:type" content="blog">
    <meta property="og:title" content="a simple gene finder">
    <meta property="og:url" content="/2011/07/a-simple-gene-finder/">
    <meta property="og:site_name" content="Guangchuang Yu">
    <meta name="twitter:card" content="summary">
    <meta name="twitter:title" content="Guangchuang Yu">
    <meta name="twitter:description" content="ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œhttp://ibw2011.fmmu.edu.cn/schedule.htm
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚

è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚


Course Projects:

Project 1: Implementation of a simple gene finder

GOAL

Build a simple codon-usage based gene finder for finding genes in
E.coli.

Procedure

Collect 100 gene sequences from the bacterium E. coli in the genbank
(http://www.ncbi.nlm.nihh.gov). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:

Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile

Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.

The output should be printed to the standard output as (xxx stands for
the likelihood)

ORF1: xxx ORF2: xxx


">
    
      <meta name="twitter:creator" content="@guangchuangyu">
    
    

    
    

    
      <meta property="og:image" content="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">
    

    
    
    

    

    
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.css" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" />
    
    
    <link rel="stylesheet" href="https://guangchuangyu.github.io/cn/css/style-u6mk0ojoywresbx8iepslrmmhl4stuhrsxuwhkpwrkrx7mryjcaimasnk4pi.min.css" />
    
    

    
      
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-77901140-1', 'auto');
ga('send', 'pageview');
</script>

    
    
  </head>

<body>
    <div id="blog">
        <header id="header" data-behavior="4">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="https://guangchuangyu.github.io/cn/">Guangchuang Yu</a>
  </div>
  
    
      <a class="header-right-picture "
         href="https://guangchuangyu.github.io/cn/#about">
    
    
    
      
        <img class="header-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=90" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

        <nav id="sidebar" data-behavior="4">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <a href="https://guangchuangyu.github.io/cn/#about">
          <img class="sidebar-profile-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Guangchuang Yu</h4>
        
        <h5 class="sidebar-profile-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</h5>
        <img alt="å¾®ä¿¡å…¬ä¼—å·" src="https://guangchuangyu.github.io/blog_images/biobabble.jpg" width="200" height="200">
        
      </div>
      
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/categories">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/tags">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/archives">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>

    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://github.com/GuangchuangYu" target="_blank">
    
      <i class="sidebar-button-icon fa fa-lg fa-github"></i>
      
      <span class="sidebar-button-desc">GitHub</span>
    </a>
  </li>

    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/index.xml">
    
      <i class="sidebar-button-icon fa fa-lg fa-rss"></i>
      
      <span class="sidebar-button-desc">RSS</span>
    </a>
  </li>

    </ul>
  </div>
</nav>

        

        <div id="main" data-behavior="4"
             class="
                 hasCoverMetaIn
             ">
            <article class="post" itemscope itemType="http://schema.org/BlogPosting">
                
                
                <div class="post-header main-content-wrap text-left">
  
    <h1 class="post-title" itemprop="headline">
      a simple gene finder
    </h1>
  
  <div class="postShorten-meta post-meta">
  
    <time itemprop="datePublished" datetime="2011-07-10T09:04:46&#43;08:00">
      
  July 10, 2011

    </time>
  
  
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/cn/categories/conference">conference</a>, 
    
      <a class="category-link" href="https://guangchuangyu.github.io/cn/categories/r">R</a>
    
  


</div>

</div>
                
                <div class="post-content markdown" itemprop="articleBody">
                    <div class="main-content-wrap">
                        <p>ä¸Šäº†å‡ å¤©çš„è¯¾ï¼Œ<a href="http://ibw2011.fmmu.edu.cn/schedule.htm">http://ibw2011.fmmu.edu.cn/schedule.htm</a>
ä»Šå¤©å°±ä¸Šå®Œäº†ï¼Œåªå®Œæˆäº†project 1ï¼Œæƒ³å†™gibbs samplingï¼Œä½†æ˜¯æ²¡ææ˜ç™½ï¼Œæ±—ã€‚</p>

<p>è¿™ä¸ªçº¯å±ç»ƒä¹ ç”¨ï¼Œæ²¡å•¥å®ç”¨ä»·å€¼ã€‚</p>

<blockquote>
<p>Course Projects:</p>

<p>Project 1: Implementation of a simple gene finder</p>

<p>GOAL</p>

<p>Build a simple codon-usage based gene finder for finding genes in
E.coli.</p>

<p>Procedure</p>

<p>Collect 100 gene sequences from the bacterium E. coli in the genbank
(<a href="http://www.ncbi.nlm.nihh.gov">http://www.ncbi.nlm.nihh.gov</a>). Compute the codon usage table based on
these genes (and the translated protein sequences from them); Build a
probabilistic model based on the codon usages; Implement a random
sequence model in which the nucleotide frequency is computed from the
100 E. coli genes. For a given DNA sequence (and one selected reading
frame), compare your model with a random sequence model; Results that
you should submit:</p>

<p>Two FASTA files for the collected 100 genes and 100 translated protein
sequences; The printed codon usage table; A program named ECgnfinder,
running with the syntax as ECgnfinder â€“i inputfile</p>

<p>Inputfile stands for the name of input file, which should contain one
DNA sequence in FASTA file format; the program should be able to
report an error message if the input file is in the wrong format.</p>

<p>The output should be printed to the standard output as (xxx stands for
the likelihood)</p>

<p>ORF1: xxx ORF2: xxx</p>
</blockquote>

<p></p>

<pre><code class="language-r">
###########################################################
##                                                       ##
##  Dragonstar Course Project (IBW 2011)                 ##
##                                                       ##
##  Project 1: Implementation of a simple gene finder    ##
##  Author Guangchuang your                              ##
##  Xi'an (2011-07-08)                                   ##
##                                                       ##
###########################################################



##' .. content for \description{} (ECgnfinder: An R program to predict prokaryotic genes (e.g. E.coli) using 1st Markov Chain) ..
##'
##' .. content for \details{} ..
##' @title ECgnfinder
##' @param traningSequence : A FASTA file containing multiple sequences for training set
##' @param testingSequence : A FASTA file containing multiple sequences for testing
##' @param backgroundSequence : A FASTA file containing multiple sequences for building the background model
##' @return log likelihood ratio
##' @author Guangchuang Yu
ECgnfinder &lt;- function(trainingSequence, testingSequence, backgroundSequence, print=FALSE) {
    print (&quot;reading training sequence file...&quot;)
    trainingFasta &lt;- readFASTA(trainingSequence)
    trainSeq &lt;- sapply(trainingFasta, function(i) i$seq)

    print(&quot;building Transition Probability Matrix...&quot;)
    TPM &lt;- getTPM(trainSeq)

    print(&quot;building codon usage table...&quot;)
    codonUsage &lt;- getCodonUsage(trainSeq)

    print(&quot;building NucleoTide table...&quot;)
    NtUsage &lt;- getNtUsage(backgroundSequence)


    print(&quot;reading testing sequence file...&quot;)
    testing &lt;- readFASTA(testingSequence)
    testingSeq &lt;- sapply(testing, function(i) i$seq)

    x &lt;- lapply(testingSeq, logLikelihood, transition=TPM, codonusage=codonUsage, background=NtUsage)
    names(x) &lt;- paste(&quot;Sequence&quot;, seq_along(testingSeq), sep=&quot;&quot;)
    if (print) {
        x
    }
    return(x)
}


##' .. content for \description{} (read FASTA file) ..
##'
##' .. content for \details{} ..
##' @title readFASTA
##' @param file: A FASTA file
##' @return a list of list which containing desc and seq.
##' @author Guangchuang Yu
readFASTA &lt;- function(file) {
    x=scan(file, what=&quot;&quot;, sep=&quot;\n&quot;)
    idx &lt;- grep(&quot;&gt;&quot;, x)
    fasta &lt;- list()
    for (i in 1:length(idx)) {
            desc &lt;- x[idx[i]]
            if (i == length(idx)) {
                stop &lt;- length(x)
            } else {
                stop &lt;- idx[i+1]-1
            }
            seq.idx &lt;- seq(idx[i]+1, stop)
            s &lt;- x[seq.idx]
            s &lt;- paste(s, collapse=&quot;&quot;)
            ss &lt;- list(desc=desc, seq=s)
            fasta[i] &lt;- list(ss)
    }
    return(fasta)
    }


##' Build a transition probability matrix from the given Sequence set.
##'
##' @title getTPM
##' @param sequences : A vector of Sequences.
##' @return transition probability matrix
##' @author Guangchuang Yu
getTPM &lt;- function(sequences) {
    codonLength &lt;- 3
    codons &lt;- getCodon()
    
    ## calculate a transition count matrix
    transition &lt;- matrix(1, nrow=length(codons), ncol=length(codons))  ## pseudo count
    colnames(transition) &lt;- codons
    rownames(transition) &lt;- codons

    for (x in 1:length(sequences)) {
        s &lt;- sequences[x]
        len &lt;- nchar(s)
        ## drop the last alphabets which is too short (

        idx &lt;- seq(1, len, codonLength)
        for (i in seq_along(idx)) {
            ## codonI--codonJ
            codonI &lt;- substr(s,idx[i], idx[i]+codonLength-1)
            if (i != length(idx)) {
                codonJ &lt;- substr(s,idx[i+1], idx[i+1]+codonLength-1)
            }
            if (codonI %in% codons &amp; codonJ %in% codons) {
                transition[codonI, codonJ] &lt;- transition[codonI, codonJ] +1
            } else {
                warning(&quot;codon &quot;, codonI, &quot; or &quot;, codonJ, &quot; which does not defined was detected...&quot;)
            }
        }
    }

    ## Calculate Transition Probability Matrix..
    rowsum &lt;- rowSums(transition)
    TPM &lt;- transition/rowsum
    return(TPM)
}

getCodon &lt;- function() {
    codons &lt;- c(&quot;TTT&quot;, &quot;TTC&quot;, &quot;TTA&quot;, &quot;TTG&quot;,
                &quot;CTT&quot;, &quot;CTC&quot;, &quot;CTA&quot;, &quot;CTG&quot;,
                &quot;ATT&quot;, &quot;ATC&quot;, &quot;ATA&quot;, &quot;ATG&quot;,
                &quot;GTT&quot;, &quot;GTC&quot;, &quot;GTA&quot;, &quot;GTG&quot;,
                &quot;TCT&quot;, &quot;TCC&quot;, &quot;TCA&quot;, &quot;TCG&quot;,
                &quot;CCT&quot;, &quot;CCC&quot;, &quot;CCA&quot;, &quot;CCG&quot;,
                &quot;ACT&quot;, &quot;ACC&quot;, &quot;ACA&quot;, &quot;ACG&quot;,
                &quot;GCT&quot;, &quot;GCC&quot;, &quot;GCA&quot;, &quot;GCG&quot;,
                &quot;TAT&quot;, &quot;TAC&quot;, &quot;TAA&quot;, &quot;TAG&quot;,
                &quot;CAT&quot;, &quot;CAC&quot;, &quot;CAA&quot;, &quot;CAG&quot;,
                &quot;AAT&quot;, &quot;AAC&quot;, &quot;AAA&quot;, &quot;AAG&quot;,
                &quot;GAT&quot;, &quot;GAC&quot;, &quot;GAA&quot;, &quot;GAG&quot;,
                &quot;TGT&quot;, &quot;TGC&quot;, &quot;TGA&quot;, &quot;TGG&quot;,
                &quot;CGT&quot;, &quot;CGC&quot;, &quot;CGA&quot;, &quot;CGG&quot;,
                &quot;AGT&quot;, &quot;AGC&quot;, &quot;AGA&quot;, &quot;AGG&quot;,
                &quot;GGT&quot;, &quot;GGC&quot;, &quot;GGA&quot;, &quot;GGG&quot;
                )
    return(codons)
}


##' .. content for \description{} (estimate codon usage table based on training sequences) ..
##'
##' .. content for \details{} ..
##' @title getCodonUsage
##' @param sequences : A vector of sequence set.
##' @return A data.frame containing two columns of codons and Freq (frequency probabilities)
##' @author Guangchuang Yu
getCodonUsage &lt;- function(sequences) {
    codons &lt;- getCodon()
    codon.table &lt;- data.frame(codons=codons, Freq=rep(0, length(codons)))
    rownames(codon.table) = codons

    for (i in 1:length(sequences)) {
        codon &lt;- codonCount(sequences[i])
        rowname.idx &lt;- as.character(codon[,1])
        codon.table[rowname.idx,2] &lt;- codon.table[rowname.idx,2] + codon[,2]
    }
    codon.table[,2] &lt;- codon.table[,2]/sum(codon.table[,2])
                                        #print(codon.table)  #### print result.
    return(codon.table)
}

##' .. content for \description{} (no empty lines) ..
##'
##' .. content for \details{} ..
##' @title codonCount
##' @param seqStr : A sequence string
##' @return A data.frame containing two columns of codons and their counts
##' @author Guangchuang Yu
codonCount &lt;- function(seqStr) {
    codonLength=3
    len &lt;- nchar(seqStr)
    ## drop the last alphabets which is too short (

    idx &lt;- seq(1, len,codonLength)
    codons &lt;- c()
    for ( i in 1:length(idx)) {
        codon &lt;- substr(seqStr, idx[i], idx[i]+codonLength-1)
        codons &lt;- c(codons, codon)
    }
    codons.df &lt;- as.data.frame(table(codons))
    return(codons.df)
}

##' .. content for \description{} (reverse complementary sequence) ..
##'
##' .. content for \details{} ..
##' @title revcom
##' @param seqStr : A sequence string
##' @return A reverse complementary sequence
##' @author Guangchuang Yu
revcom &lt;- function(seqStr) {
    seqStr.str &lt;- unlist(strsplit(seqStr, split=&quot;&quot;))
    idxT &lt;- which(seqStr.str==&quot;T&quot;)
    idxG &lt;- which(seqStr.str==&quot;G&quot;)
    idxA &lt;- which(seqStr.str==&quot;A&quot;)
    idxC &lt;- which(seqStr.str==&quot;C&quot;)
    seqStr.str[idxT] &lt;- &quot;A&quot;
    seqStr.str[idxA] &lt;- &quot;T&quot;
    seqStr.str[idxG] &lt;- &quot;C&quot;
    seqStr.str[idxC] &lt;- &quot;G&quot;
    seqStr.rc &lt;- rev(seqStr.str)
    seqStr.rc &lt;- paste(seqStr.rc, collapse=&quot;&quot;)
    return(seqStr.rc)
}

##' .. content for \description{} (calculate NT frequency probabilities of a given sequence set) ..
##'
##' .. content for \details{use to estimate the background/random NT usage} ..
##' @title getNtUsage
##' @param file : A FASTA file
##' @return A data.frame containing NTs and their corresponding frequency probabilities.
##' @author Guangchuang Yu
getNtUsage &lt;- function(file) {
    print(&quot;reading FASTA file&quot;)
    fasta &lt;- readFASTA(file)
    sequences &lt;- sapply(fasta, function(i) i$seq)
    x &lt;- sapply(sequences, strsplit, split=&quot;&quot;)
    x &lt;- unlist(x)
    NTprob &lt;- as.data.frame(table(x))
    NTprob[,2] &lt;- NTprob[,2]/sum(NTprob[,2])
    rownames(NTprob) &lt;- NTprob[,1]
    return(NTprob)
}

##' .. content for \description{} (calculate the log likelihood of the six possible reading frame of a given sequence) ..
##'
##' .. content for \details{} ..
##' @title logLikelihood
##' @param sequence : A sequence string
##' @param transition : Transition Probatility Matrix
##' @param codonusage : Codon Usage data.frame
##' @param background : NT Usage data.frame for background/random model
##' @return Log likelihood
##' @author Guangchuang Yu
logLikelihood &lt;- function(sequence, transition, codonusage, background) {
    codonLength &lt;- 3
    orf1 &lt;- sequence
    orf2 &lt;- substr(sequence, 2, nchar(sequence))
    orf3 &lt;- substr(sequence, 3, nchar(sequence))
    seq.rc &lt;- revcom(sequence)
    orf4 &lt;- seq.rc
    orf5 &lt;- substr(seq.rc, 2, nchar(seq.rc))
    orf6 &lt;- substr(seq.rc, 3, nchar(seq.rc))
    ORF &lt;- c(orf1, orf2, orf3, orf4, orf5, orf6)
    names(ORF) &lt;- c(&quot;ORF1&quot;, &quot;ORF2&quot;, &quot;ORF3&quot;, &quot;ORF4&quot;, &quot;ORF5&quot;, &quot;ORF6&quot;)

    result &lt;- data.frame(ORF=names(ORF), logLikelihood=rep(0,6))
    for (j in seq_along(ORF)) {
        firstCodon &lt;- substr(ORF[j], 1, 3)
        if (firstCodon %in% codonusage[,1]) {
            initProb &lt;- log(codonusage[firstCodon,2])
        } else {
            initProb &lt;- log(1/64)
        }

        transitionCount &lt;- matrix(0, nrow=nrow(transition), ncol=ncol(transition))
        rownames(transitionCount) &lt;- rownames(transition)
        colnames(transitionCount) &lt;- colnames(transition)

        codons &lt;- rownames(transition)

        len &lt;- nchar(ORF[j])
        len &lt;- len - len %% codonLength - codonLength

        idx &lt;- seq(1, len, codonLength)
        for (i in seq_along(idx)) {
            ## codonI--codonJ
            codonI &lt;- substr(ORF[j],idx[i], idx[i]+codonLength-1)
            if (i != length(idx)) {
                codonJ &lt;- substr(ORF[j],idx[i+1], idx[i+1]+codonLength-1)
            }
            if (codonI %in% codons &amp; codonJ %in% codons) {
                transitionCount[codonI, codonJ] &lt;- transitionCount[codonI, codonJ] +1
            } else {
                warning(&quot;codon &quot;, codonI, &quot; or &quot;, codonJ, &quot; which does not defined was detected...&quot;)
            }
        }

        postTransition &lt;- log(transition) * transitionCount
        posterior &lt;- initProb + sum(postTransition)

        NTseq &lt;- unlist(strsplit(ORF[j], split=&quot;&quot;))
        orf.len &lt;- nchar(ORF[j]) - nchar(ORF[j]) %% codonLength
        NTseq &lt;- NTseq[1:orf.len]
        randomModel &lt;- sum(log(background[NTseq,2]))

        likelihood &lt;- posterior - randomModel
        result[j, 2] &lt;- likelihood
    }
    return(result)
}
</code></pre>
                        
                        <center><span class="zaln"><a href="https://guangchuangyu.github.io/blog_images/pay_qrcode.png" class="zb">èµèµ</a></span></center>
                        <center><span class="zaln"><a href="https://guangchuangyu.github.io/cn/2017/11/bmc-journal-protects-paper-of-plagiarism/" class="zb">æŠµåˆ¶BMCï¼ŒåŒ¹å¤«æœ‰è´£</a></span></center>
                    </div>
                </div>
                <div id="post-footer" class="post-footer main-content-wrap">
                    
                    
                    
                    
                    <div class="post-footer-tags">
                        <span class="text-color-light text-small">TAGGED IN</span><br/>
                        
  <a class="tag tag--primary tag--small" href="https://guangchuangyu.github.io/cn//tags/ibw/">IBW</a>

                    </div>
                    
                    
                    
                    <div class="post-actions-wrap">
  <nav>
    <ul class="post-actions post-action-nav">
      
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/07/ibw2011/" data-tooltip="IBW2011">
          
            <i class="fa fa-angle-left"></i>
            <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
          </a>
        </li>
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/06/screen-shot/" data-tooltip="screen shot">
          
            <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
            <i class="fa fa-angle-right"></i>
          </a>
        </li>
      
    </ul>
  </nav>
  <ul class="post-actions post-action-share">
    
      <li class="post-action hide-lg hide-md hide-sm">
        <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
          <i class="fa fa-share-alt"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-google-plus"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-facebook-official"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-twitter"></i>
        </a>
      </li>
    
    
      <li class="post-action">
        <a class="post-action-btn btn btn--default" href="#disqus_thread">
          <i class="fa fa-comment-o"></i>
        </a>
      </li>
    
    <li class="post-action">
      
        <a class="post-action-btn btn btn--default" href="#">
      
        <i class="fa fa-list"></i>
      </a>
    </li>
  </ul>
</div>


                    
                    
                    <div id="disqus_thread">
  <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
                    
                    
                </div>
            </article>
            <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2018 Guangchuang Yu. All Rights Reserved
  </span>
</footer>

        </div>
        <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
            <div class="post-actions-wrap">
  <nav>
    <ul class="post-actions post-action-nav">
      
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/07/ibw2011/" data-tooltip="IBW2011">
          
            <i class="fa fa-angle-left"></i>
            <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
          </a>
        </li>
        <li class="post-action">
          
            <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2011/06/screen-shot/" data-tooltip="screen shot">
          
            <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
            <i class="fa fa-angle-right"></i>
          </a>
        </li>
      
    </ul>
  </nav>
  <ul class="post-actions post-action-share">
    
      <li class="post-action hide-lg hide-md hide-sm">
        <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
          <i class="fa fa-share-alt"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://plus.google.com/share?url=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-google-plus"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-facebook-official"></i>
        </a>
      </li>
      <li class="post-action hide-xs">
        <a class="post-action-btn btn btn--default" target="new" href="https://twitter.com/intent/tweet?text=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
          <i class="fa fa-twitter"></i>
        </a>
      </li>
    
    
      <li class="post-action">
        <a class="post-action-btn btn btn--default" href="#disqus_thread">
          <i class="fa fa-comment-o"></i>
        </a>
      </li>
    
    <li class="post-action">
      
        <a class="post-action-btn btn btn--default" href="#">
      
        <i class="fa fa-list"></i>
      </a>
    </li>
  </ul>
</div>


        </div>
        <div id="share-options-bar" class="share-options-bar" data-behavior="4">
  <ul class="share-options">
    <li class="share-option">
      <a class="share-option-btn" target="new" href="https://plus.google.com/share?url=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
        <i class="fa fa-google-plus"></i><span>Share on Google Plus</span>
      </a>
    </li>
    <li class="share-option">
      <a class="share-option-btn" target="new" href="https://www.facebook.com/sharer/sharer.php?u=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
        <i class="fa fa-facebook-official"></i><span>Share on Facebook</span>
      </a>
    </li>
    <li class="share-option">
      <a class="share-option-btn" target="new" href="https://twitter.com/intent/tweet?text=https%3a%2f%2fguangchuangyu.github.io%2fcn%2f2011%2f07%2fa-simple-gene-finder%2f">
        <i class="fa fa-twitter"></i><span>Share on Twitter</span>
      </a>
    </li>
  </ul>
</div>
<div id="share-options-mask" class="share-options-mask"></div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Guangchuang Yu</h4>
    
      <div id="about-card-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Postdoc researcher
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        Hong Kong
      </div>
    
  </div>
</div>

    <div id="algolia-search-modal" class="modal-container">
  <div class="modal">
    <div class="modal-header">
      <span class="close-button"><i class="fa fa-close"></i></span>
      <a href="https://algolia.com" target="_blank" class="searchby-algolia text-color-light link-unstyled">
        <span class="searchby-algolia-text text-color-light text-small">by</span>
        <img class="searchby-algolia-logo" src="https://www.algolia.com/static_assets/images/press/downloads/algolia-light.svg">
      </a>
      <i class="search-icon fa fa-search"></i>
      <form id="algolia-search-form">
        <input type="text" id="algolia-search-input" name="search"
          class="form-control input--large search-input" placeholder="Search" />
      </form>
    </div>
    <div class="modal-body">
      <div class="no-result text-color-light text-center">no post found</div>
      <div class="results">
        
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/02/broken-heart/">
                <h3 class="media-heading">ä¸€åƒä¸ªä¼¤å¿ƒçš„ç†ç”±</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Feb 2, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>çœ¼çœ‹ç€2æœˆ14å·å°±è¦åˆ°æ¥äº†ï¼Œåˆæ˜¯ä¸€å¹´ä¸€åº¦çš„ç”»çº¢å¿ƒè£…13çš„æ—¶å€™ï¼Œè¿™ç§äº‹æƒ…å·²ç»è¢«æˆ‘BSäº†ï¼Œè¯·çŒ›å‡»ï¼šã€Š<a href="http://mp.weixin.qq.com/s/Y8IHd6X4f49irXvD_M7bNw">ç”»â¤ï¸è£…geekä»€ä¹ˆçš„ï¼Œéƒ½å¼±çˆ†äº†ï¼ğŸ˜œ</a>ã€‹ã€‚çº¢å¿ƒå·²ç»æ²¡ä»€ä¹ˆå¥½ç”»çš„äº†ï¼Œç‰¹åˆ«æ˜¯åœ¨æˆ‘ç”»äº†ç”Ÿæ—¥è›‹ç³•ï¼Œå¹¶ç”¨è›‹ç³•ç”»çº¢å¿ƒä¹‹åï¼Œã€Š<a href="https://mp.weixin.qq.com/s/Iqcqo5YTy1qmXdsuIrlrYQ">130å²äº†ï¼Œç¥ä½ ç”Ÿå¿«</a>ã€‹ï¼Œæ˜¯é©´å­æ˜¯é©¬ï¼Œéƒ½æ²¡æ³•æ™’è‚Œè‚‰äº†ã€‚å½“ç„¶çº¢å¿ƒè¿˜æ˜¯ä¼šæœ‰æ‰€è°“çš„ç†å·¥å®…ç”·æ™’ï¼Œé™¤æ­¤ä¹‹å¤–è¿˜ä¼šæœ‰å„ç§èŠ±æ‹›ï¼Œæ¥ç»™å•èº«ç‹—æ’’ç‹—ç²®ã€‚æƒ³è¦åŠ å…¥æ’’ç‹—ç²®å¤§å†›çš„ï¼Œå¿«ç‚¹çœ‹ã€Š<a href="http://mp.weixin.qq.com/s/2j9VRgar5ExSma6B7Yc3dw">å¦‚ä½•å‘Šåˆ«å•èº«</a>ã€‹ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/02/job-wanted/">
                <h3 class="media-heading">Yå”è¦æ±‚èŒ</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Feb 2, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>ä½™å…‰åˆ›
é¦™æ¸¯å¤§å­¦ ç”Ÿç‰©ä¿¡æ¯å­¦ åšå£«ï¼Œç›®å‰åœ¨æ¸¯å¤§ä»äº‹åšå£«åå·¥ä½œ
æ±‚èŒæ„å‘ï¼šå‰¯æ•™æˆä»¥ä¸Š
åœ°ç‚¹ï¼šä¸»è¦æ„å‘æ˜¯å¹¿å·å’Œæ·±åœ³</p>

<p>ç®€å†ï¼š<a href="https://guangchuangyu.github.io/resume/YGC.pdf">https://guangchuangyu.github.io/resume/YGC.pdf</a></p>

<p>google scholaræŒ‡æ ‡ï¼š</p>

<ul>
<li>å¼•ç”¨ï¼š1330</li>
<li>h-index: 14</li>
<li>i10-index: 17</li>
</ul>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/02/wechat-addition/">
                <h3 class="media-heading">ç½‘ç˜¾ä½ å¥½ï¼Œæˆ‘æ˜¯80åï¼Œè¯·å†²æˆ‘æ¥</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Feb 2, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p><img src="https://guangchuangyu.github.io/blog_images/2018/wechat_additive/images.png" alt="" /></p>

<p>ç«¥é‹ä»¬ï¼Œæ•²é»‘æ¿åˆ’é‡ç‚¹å•¦ï¼Œä¸Šå›¾è¯·å¯¹å·å…¥åï¼Œæ²¡é”™ï¼Œè¯´çš„å°±æ˜¯ä½ ï¼è¿™ä¸ªç½‘ç˜¾çš„éªšå¹´ï¼</p>

<p>Scientific Reportæœ€è¿‘å‘è¡¨äº†<a href="https://www.nature.com/articles/s41598-018-19904-y">ä¸€ç¯‡ç¥æ–‡</a>:</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/01/scihub-bookmark/">
                <h3 class="media-heading">å…³äºscihubï¼Œä¸€æ–‡å°±å¤Ÿäº†</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jan 1, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>Sci-Hubæ˜¯å±…å®¶æ—…è¡Œå­¦æœ¯å¿…å¤‡çš„ç¥å™¨ï¼Œæ¯•ç«Ÿç©·äººæ¯”è¾ƒå¤šï¼Œè€Œå‡ºç‰ˆç¤¾å¤ªé»‘å¿ƒã€‚ä½†Sci-Hubæœ‰ä¸€ä¸ªé—®é¢˜æ˜¯é•œåƒç«™ç»å¸¸æŒ‚ï¼Œæˆ‘ä¹‹å‰ç»™å‡ºäº†ä¸€ä¸ªæ¯”è¾ƒå¯èƒ½ä¸ä¼šæŒ‚çš„é“¾æ¥ï¼Œã€Š<a href="https://mp.weixin.qq.com/s/5JGC1bcVZOmdaPxo_uXYYw">å†ä¸ç”¨æ‹…å¿ƒæ‰“ä¸å¼€sci-hubäº†</a>ã€‹ï¼Œé‚£å°±æ˜¯æ´‹è‘±é“¾æ¥ã€‚å½“ç„¶æ´‹è‘±é“¾æ¥æ¯”è¾ƒæ…¢ï¼Œæˆ‘ä»¬è¿˜æ˜¯æ›´å€¾å‘äºä½¿ç”¨æ™®é€šçš„é“¾æ¥ï¼Œæ‰€ä»¥æˆ‘åˆæ”¶é›†äº†ã€Š<a href="https://mp.weixin.qq.com/s/RK1CYEboiEDzapdHsiwEGg">2018æœ€æ–°å¯ç”¨çš„sci-hubé•œåƒ+æ¡Œé¢ç‰ˆç¨‹åº</a>ã€‹ï¼Œå½“ç„¶è¿™äº›é“¾æ¥éšç€æ—¶é—´çš„æ¨ç§»ï¼Œæœ‰äº›å¯èƒ½ä¹Ÿä¼šæŒ‚ã€‚</p>

<p>ä»»ä½•æ—¶å€™ï¼Œæˆ‘ä»¬éƒ½åº”è¯¥æœ‰å¤šæ‰‹å‡†å¤‡ï¼Œæ‰€ä»¥æˆ‘ä¸€ç›´éƒ½æœ‰å¤‡é€‰æ–¹æ¡ˆï¼Œè™½ç„¶ä¸å¦‚sci-hubï¼Œè™½ç„¶ç°åœ¨sci-hubè¿˜èƒ½ç”¨çš„æƒ…å†µä¸‹é¦–å…ˆè¿˜æ˜¯sci-hubï¼Œä½†æ”¶è—ä¸€ä¸¤ä¸ªå¤‡é€‰å·¥å…·æ˜¯å¿…è¦çš„ï¼Œå®ƒè®©ä½ åœ¨é¢å‰å˜åŒ–çš„æ—¶å€™å¯ä»¥æ·¡å®šã€‚ã€Š<a href="http://mp.weixin.qq.com/s/aaIby87LSQgxpvCAECFxkw">sci-hubå¦‚æœæŒ‚äº†ï¼Œä½ è¿˜æœ‰ç¥å™¨ä¸‹æ–‡çŒ®å—ï¼Ÿ</a>ã€‹è¿™ç¯‡æ–‡ç« ä»‹ç»çš„ç¥å™¨å³ä½¿ä½ ç°åœ¨ä¸ç”¨ï¼Œä¹Ÿè¯¥æ”¶è—ç€ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/01/frog-tree/">
                <h3 class="media-heading">Yå”ä¸æƒ³å…»è›™ï¼Œåªæƒ³å…»ä½ ï¼</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jan 1, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>è¯è¯´æœ‰ä¸€åªè›™ï¼Œå®ƒæ—…æ¸¸å»äº†å¹¿ä¸œï¼Œæ¸¸æˆç»“æŸã€‚</p>

<p>è¿™ç¯‡æ–‡ç« ï¼Œåªæƒ³è·Ÿå¤§å®¶è®²ä¸€è®²ggtreeï¼Œä½ å€¼å¾—æ‹¥æœ‰çš„å¼ºå¤§çš„è¿›åŒ–æ ‘å¯è§†åŒ–è½¯ä»¶ï¼Œé«˜é¢œå€¼çš„å›¾ï¼Œå°±åœ¨äºæŒ‡å°–å‡ è¡Œä»£ç ä¸­ã€‚</p>

<p>æœ€è¿‘å…»è›™å¾ˆæµè¡Œï¼Œå°±ä»¥è›™ä¸ºä¾‹ï¼Œè¯»æ ‘ç”¨<code>read.tree</code>ï¼Œç„¶åä¸€ä¸ª<code>ggtree</code>çš„æŒ‡ä»¤ï¼Œæ ‘å°±ç”»å‡ºæ¥äº†ã€‚è€Œä½ è§è¯ç¥å¥‡çš„æ—¶åˆ»å°±åœ¨äºä½ ä¸æ–­<code>+geom_xxx</code>åŠ å›¾å±‚ï¼Œè¿™é‡Œæˆ‘ç”¨äº†<code>geom_tiplab</code>æ¥åŠ taxa labelï¼Œè€Œè¿™ä¸ªå¯ä¸ä»…ä»…æ˜¯åå­—å“¦ï¼Œç”¨emojiä¹Ÿå¯ä»¥ï¼Œè¿™é‡Œæˆ‘ç”¨çš„ç«Ÿç„¶æ˜¯<strong>å›¾ç‰‡</strong>ï¼Œä½ æ²¡çœ‹é”™ï¼Œä¹Ÿå¯ä»¥ã€‚åªè¦æŒ‡å®š<code>geom=&quot;image&quot;</code>å°±OKï¼Œé‚£ä¸ªæœ¬æ¥è¦æ‰“çš„æ–‡æœ¬ï¼Œå°±ä¼šè¢«å½“æˆæ˜¯æ–‡ä»¶åï¼Œç„¶åè¯»å›¾ç‰‡ç”»å›¾ç‰‡ï¼Œä¸€åˆ‡å°½åœ¨æˆ‘çš„ggimageåŒ…ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/01/scihub-osx/">
                <h3 class="media-heading">è‹¹æœç‰ˆSciHubæ¡Œé¢è½¯ä»¶</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jan 1, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>ä¸ºäº†æ–¹ä¾¿è‹¹æœç”¨æˆ·ä¹Ÿèƒ½ç”¨æ¡Œé¢è½¯ä»¶ä¸‹æ–‡çŒ®ï¼Œæˆ‘æŠŠSciHubæ¡Œé¢ç‰ˆæ‰“åŒ…æˆå¯ä»¥è¿è¡Œåœ¨è‹¹æœä¸Šçš„Appã€‚</p>

<p>SciHubæ¡Œé¢ç‰ˆæ˜¯Windowsç¨‹åºï¼Œä¸»é¡µä¸º<a href="https://zhuanlan.zhihu.com/p/31809890">https://zhuanlan.zhihu.com/p/31809890</a>ã€‚</p>

<p>é¦–å…ˆè¿™æ˜¯åˆ©ç”¨wineæŠ€æœ¯å®ç°çš„ï¼Œåœ¨è‹¹æœä¸Šä½ éœ€è¦å®‰è£…<strong>XQuartz</strong>: <a href="https://www.xquartz.org/">https://www.xquartz.org/</a>ã€‚</p>

<p>å®‰è£…å¾ˆå®¹æ˜“ï¼Œæˆ‘å·²ç»æ‰“åŒ…äº†dmgæ–‡ä»¶ï¼Œè¿™ä¸ªæ–‡ä»¶æœ‰44Mï¼Œè¿˜æŒºå¤§çš„ï¼Œæ²¡æ³•ï¼Œè¦æ‰“åŒ…ä¸€ä¸ªwindowsçš„è¿è¡Œç¯å¢ƒå˜›ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/01/pubmed-still-update/">
                <h3 class="media-heading">PubMedæ²¡æœ‰åœæ›´</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jan 1, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>åœ¨ä¸€ç¯‡æ–‡ç« ã€Š<a href="http://mp.weixin.qq.com/s/VfLmahpFqdmXfwId-d_BWQ">pubmedåœæ›´ï¼Ÿå’Œä½ æ²¡åŠæ¯›é’±å…³ç³»ï¼ä½ ä»¬æ˜¯æƒ³æ”¾å‡æƒ³ç–¯äº†å§ï¼Ÿ</a>ã€‹æˆ‘å°±è§£è¯»äº†æ‰€è°“çš„åœæ›´ï¼Œæ ¹æœ¬ä¸ç¢äº‹ï¼Œç„¶ååˆåˆ†æäº†äº‹æƒ…åªæ˜¯è¢«è‡ªåª’ä½“æ‹¿æ¥åšçœ¼çƒè€Œå·²ï¼ŒçœŸæ­£å—å½±å“çš„NCBIæ•°æ®åº“æ²¡äººè¯´ï¼Œè€ŒPubMedåœæ›´çš„å¯èƒ½æ€§ä¸å¤§ã€‚</p>

<blockquote>
<p>Because of a lapse in government funding, the information on this website may not be up to date, transactions submitted via the website may not be processed, and the agency may not be able to respond to inquiries until appropriations are enacted.</p>
</blockquote>

<p>è¿™ä¸ªæ¶ˆæ¯ï¼Œå¾ˆå¤šäººä¸Špubmedçœ‹åˆ°ï¼Œå°±æ‰“é¸¡è¡€äº†ï¼Œå“Ÿå“Ÿå“Ÿï¼Œå·å¤–å·å¤–å¤§æ–°é—»ï¼ŒçœŸçš„æœ‰è¿‡è¿‡è„‘å­å—ï¼Ÿ</p>

<blockquote>
<p>transactions submitted via the website may not be processed</p>
</blockquote>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/01/ggplot2-seed/">
                <h3 class="media-heading">ggplot2çš„ä¸€ä¸ªå‘</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jan 1, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>ä»Šå¤©ä»¥ggplot2çš„ä¸€ä¸ªå‘æ¥è¯´ä¸€ä¸‹ï¼Œå‘æ— å¤„ä¸åœ¨ï¼Œé˜²ä¸èƒœé˜²ï¼Œä½ å¤§å¯ä»¥è¯•ä¸€ä¸‹ä¸‹é¢çš„ä»£ç ï¼š</p>

<pre><code>&gt; set.seed(123)
&gt; require(ggplot2)
Loading required package: ggplot2
&gt; rnorm(3)
[1]  0.8005543  1.1902066 -1.6895557
&gt; set.seed(123)
&gt; rnorm(3)
[1] -0.5604756 -0.2301775  1.5587083
</code></pre>

<p>åœ¨ä¸¤æ¬¡set.seedå’Œrnormä¹‹é—´ï¼Œç¬¬ä¸€æ¬¡å› ä¸ºåŠ è½½äº†ggplot2ï¼Œç»“æœå°±ä¸ä¸€æ ·äº†ï¼è¿™å¿…é¡»æ˜¯ç¬¬äºŒæ¬¡æ˜¯æ­£ç¡®ç­”æ¡ˆï¼Œä¹Ÿå°±æ˜¯è¯´åŠ è½½ggplot2æŠŠä½ çš„seedç»™åƒäº†ï¼åŠ è½½åŒ…ä¼šæ”¹å˜Rç¯å¢ƒï¼Ÿè¿™ç»å¯¹ä¸æ˜¯å¥½ä¸»æ„ï¼Œæˆ‘ä»¬æ¥è¯•è¯•åŠ è½½åˆ«çš„åŒ…è¯•è¯•ï¼Œæ¯”å¦‚æˆ‘çš„clusterProfiler:</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/01/pubmed-stop-update/">
                <h3 class="media-heading">PubMedåœæ›´ï¼Ÿ</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jan 1, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>ä»Šå¤©å­¦æœ¯åœˆåˆ·å±çš„æ˜¯ç¾å›½æ”¿åºœå…³é—¨ï¼Œpubmedåœæ›´ï¼Œæœ‰äº›æ ‡é¢˜å…šè¯´æˆäº†ç¾å›½æ”¿åºœç ´äº§ï¼Œåˆè¯´å¾—å¥½åƒä¸­å›½äººçš„ç§‘ç ”ä¹Ÿè¦è·Ÿç€æ”¾å‡ä¼¼çš„ï¼Œè¿˜ä¸€å †äººç–¯ä¼ ï¼Œä¼¼ä¹æš—ç¤ºç€pubmedä¹Ÿå…³é—¨ï¼Œè€Œä¸”æ²¡æœ‰pubmedä¼šæ­»ä¸€æ ·ã€‚æ‰€ä»¥æˆ‘æ„Ÿè§‰æœ‰å¿…è¦æ™®åŠä¸€ä¸‹å§¿åŠ¿ï¼Œæ¯•ç«Ÿåƒç“œçš„äººç¾¤åŸºæ•°å¤ªå¤§ã€‚</p>

<blockquote>
<p>Holy shit what a fucking disaster, pubmed is such an amazing resource, unbelievable the US would let it waste away. What an embarrassment of a country.</p>
</blockquote>

<p>æ°”æ„¤å½’æ°”æ„¤ï¼Œä½†æ ‡é¢˜å…šç”šè‡³äºè·‘åäº†å°±ä¸å¯å–äº†ã€‚æ— éåªæ˜¯åœæ›´ï¼Œä¹Ÿå°±æ˜¯è¯´æ–‡ç« è¯¥å‘çš„ç…§æ ·å‘ï¼Œåªæ˜¯æš‚æ—¶ä¸ä¼šè¢«æ”¶å½•è¿›pubmedè€Œå·²ï¼Œä¹Ÿå°±æ˜¯è¯´pubmedç…§å¸¸ç”¨ï¼Œåªä¸è¿‡æœä¸åˆ°æœ€æ–°çš„æ–‡çŒ®ï¼Œå¦‚æœè¯´è¿™æ ·å°±å¯ä»¥å»æ”¾å¤§å‡äº†ï¼Œé‚£ä¹ˆä¼¼ä¹å¦‚æœä¸åœæ›´ï¼Œä½ çœ‹å¤šäº†å‡ ç¯‡æ–°æ–‡çŒ®ï¼Œå°±èƒ½å¤Ÿå‘paperï¼Œè¿å¨¶ç™½å¯Œç¾äº†ï¼Ÿè€Œçœ‹å°‘äº†è¿™æœ€æ–°çš„æ–‡çŒ®ï¼Œå°±ç§’å˜ç§‘ç ”å±Œä¸ï¼Œå¯ä»¥å›å®¶ç¡å¤§è§‰äº†ï¼Ÿ</p>

<blockquote>
<p>The website continues to work. The shutdown will be over in a few days and everything will go back to normal. You can also use Google Scholar, Web of Science, and Scopus.</p>
</blockquote>

<p>äº‹å®ä¸Šï¼Œpubmedåªæ˜¯ä¸€ä¸ªæœæ–‡çŒ®çš„æ•°æ®åº“ï¼Œå®ƒåœæ›´ï¼Œæ–‡ç« ç…§å‘ï¼Œè€Œå¤§å®¶å¯ä»¥é€šè¿‡åˆ«çš„é€”å¾„å»è·å¾—ï¼Œå¦‚æœä½ åªæœ‰è¿™ä¸€æ¡é€”å¾„ï¼Œé‚£ä¹ˆä½ åšç§‘ç ”ä¹Ÿç¡®å®å°‘äº†ç‚¹Bæ•°ã€‚ä¸Šé¢æåˆ°çš„google scholarå¾ˆå¤šäººå¯èƒ½ä¸Šä¸å»ï¼ˆ<a href="http://mp.weixin.qq.com/s/b6xyVv-HjQQSn0pq7ILQzw">æ¢¯å­åœ¨è¿™é‡Œ</a>ï¼‰ï¼Œè€ŒScopusæ˜¯è¦æ”¶è´¹çš„ã€‚ç„¶åé€”å¾„æ˜¯å¾ˆå¤šçš„ï¼Œ</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/01/scihub/">
                <h3 class="media-heading">2018æœ€æ–°å¯ç”¨çš„sci-hubé•œåƒ&#43;æ¡Œé¢ç‰ˆç¨‹åº</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jan 1, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>ä¹‹å‰çš„ã€Š<a href="https://mp.weixin.qq.com/s/TMmsOsEn0wV9yXKOLma3Rw">å¦‚ä½•è·å–æ–‡çŒ®</a>ã€‹ä»‹ç»äº†æ–¹æ–¹é¢é¢ï¼Œä¸å•å•æ˜¯ä¸‹è½½æ–‡çŒ®ï¼Œè€Œå¯¹äºä¸‹è½½æ–‡çŒ®ï¼Œscihubå¿…é¡»æ˜¯é¦–é€‰ï¼Œè™½ç„¶æˆ‘ä»¬æœ‰å¤‡é€‰æ–¹æ¡ˆã€Š<a href="http://mp.weixin.qq.com/s/aaIby87LSQgxpvCAECFxkw">sci-hubå¦‚æœæŒ‚äº†ï¼Œä½ è¿˜æœ‰ç¥å™¨ä¸‹æ–‡çŒ®å—ï¼Ÿ</a>ã€‹ï¼Œä½†æ€»æ˜¯æ²¡scihubçˆ½ï¼Œä¹‹å‰ä»‹ç»äº†ç”¨æ´‹è‘±é“¾æ¥ï¼Œã€Š<a href="https://mp.weixin.qq.com/s/5JGC1bcVZOmdaPxo_uXYYw">å†ä¸ç”¨æ‹…å¿ƒæ‰“ä¸å¼€sci-hubäº†</a>
<a href="https://mp.weixin.qq.com/s/AGQn7Lfn_nHHrtVQyJSTmg">å¬è¯´ä½ ä¹Ÿåœ¨è¯»æ–‡çŒ®</a>ã€‹ï¼Œåœ¨å†™è¿™ç¯‡æ–‡ç« çš„æ—¶å€™ï¼Œæˆ‘è¿˜æµ‹è¯•äº†ä¸€ä¸‹ï¼Œç°åœ¨ä¾ç„¶å¯ç”¨ï¼</p>

<p>ç›®å‰æŒ‚æ‰çš„scihubé•œåƒå·²ç»å¾ˆå¤šï¼Œå¯¼è‡´å¾ˆå¤šäººè®¤ä¸ºscihubæ²¡æ³•ç”¨äº†ï¼Œå…¶å®è¿˜æœ‰å¾ˆå¤šçš„é•œåƒæ˜¯å¯ä»¥ç”¨çš„ï¼Œå½“ç„¶èƒ½ç”¨çš„ï¼Œä»¥åä¹Ÿå¯èƒ½ä¼šæŒ‚ï¼Œä½†æ–°çš„é•œåƒæˆ‘ç›¸ä¿¡è¿˜æ˜¯ä¼šä¸æ–­å‡ºç°çš„ï¼Œå¤§å®¶ä¸”è¡Œä¸”çæƒœå§ã€‚</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
      </div>
    </div>
    <div class="modal-footer">
      <p class="results-count text-medium"
         data-message-zero="no post found"
         data-message-one="1 post found"
         data-message-other="{n} posts found">
         195 posts found
      </p>
    </div>
  </div>
</div>
    
  
    
    <div id="cover" style="background-image:url('https://guangchuangyu.github.io/cn/images/cover.jpg');"></div>
  


    
<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.8.0/highlight.min.js"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.js"></script>
<script src="//cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.js"></script>


<script src="https://guangchuangyu.github.io/cn/js/script-wl33z0n6ocaypepiqrazthtivfrliqijej4rq8ek8gvrv1awftmgjuv8k4zc.min.js"></script>

<script>
$(document).ready(function() {
  hljs.configure({ classPrefix: '', useBR: false });
  $('pre.code-highlight').each(function(i, block) {
    var code = "";
    hljs.highlightAuto(block.innerText).value.split(/\r\n|\r|\n/).forEach(function(line) {
      code += "<span class=\"line\">" + line + "</span><br>";
    });
    if (code.length > 0) {
      block.innerHTML = code;  
    }
  });
  $('pre > code').each(function(i, block) {
    $(this).addClass('codeblock');
    hljs.highlightBlock(block);
  });
});
</script>

  
    
      <script>
        var disqus_config = function () {
          this.page.url = 'https:\/\/guangchuangyu.github.io\/cn\/2011\/07\/a-simple-gene-finder\/';
          
            this.page.identifier = '\/2011\/07\/a-simple-gene-finder\/'
          
        };
        (function() {
          
          
          if (window.location.hostname == "localhost") {
            return;
          }
          var d = document, s = d.createElement('script');
          var disqus_shortname = 'gcyu';
          s.src = '//' + disqus_shortname + '.disqus.com/embed.js';

          s.setAttribute('data-timestamp', +new Date());
          (d.head || d.body).appendChild(s);
        })();
      </script>
    
  





    
  </body>
</html>

