<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>跨物种计算基因相似性</title>

  
  
  <link rel="stylesheet" href="https://guangchuangyu.github.io/css/hugo-octopress.css">

  
  

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link href="https://guangchuangyu.github.io/favicon.png" rel="icon">

  
  
  

  

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.21" />

  
  
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-77901140-1', 'auto');
ga('send', 'pageview');
</script>


</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="https://guangchuangyu.github.io/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">

<ul class="main-navigation">
  
  
    
      <li><a href="https://guangchuangyu.github.io/cn/" target="_blank" title="Blog">Blog</a></li>
    
  
    
      <li><a href="https://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="https://guangchuangyu.github.io/cn/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="https://guangchuangyu.github.io/cn/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="https://guangchuangyu.github.io/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>


<div id="main">
  <div id="content">
    <div>
      <article class="hentry" role="article">
        <header>
          <h1 class="entry-title">
            跨物种计算基因相似性 
          </h1>
          <p class="meta">Dec 7, 2016 - 1 minute read -  <a href="https://guangchuangyu.github.io/cn/2016/12/%E8%B7%A8%E7%89%A9%E7%A7%8D%E8%AE%A1%E7%AE%97%E5%9F%BA%E5%9B%A0%E7%9B%B8%E4%BC%BC%E6%80%A7/#disqus_thread">Comments</a>

          

          
          
          <a class="label" href="https://guangchuangyu.github.io/categories/r">R</a>
          </p>
        </header>
        <div class="entry-content">
          
          
          

	  
<div class="a2a_kit a2a_kit_size_32 a2a_default_style">
<a class="a2a_dd" href="//www.addtoany.com/share"></a>
<a class="a2a_button_facebook"></a>
<a class="a2a_button_twitter"></a>
<a class="a2a_button_google_plus"></a>
<a class="a2a_button_pinterest"></a>
<a class="a2a_button_reddit"></a>
<a class="a2a_button_sina_weibo"></a>
<a class="a2a_button_wechat"></a>
<a class="a2a_button_douban"></a>
</div>
<script async src="//static.addtoany.com/menu/page.js"></script>

<br>

    <blockquote>
<p>Hi,</p>

<p>I know this question has been asked before a long time ago and I don&rsquo;t see an answer of that question in the mailing list or in the vignette of GOsemsim package. So I was wondering what is the easiest possible way of calculating GO semantic similarity value for orthologus gene pairs between two species using the above R package or any other package you know of. I am not doing this for less annotated species I need to calculate that for orthologus genes between Human and Mouse (both of which are well annotated IMHO). So I would much appreciate it if anyone who has already done this before can point me to a resource which already has pre-calculated semantic similarity values for Mouse and Human orthologues or has inbuilt code to do that.</p>

<p>Thanks &amp; regards</p>
</blockquote>

<p>这是Bioconductor support site上的<a href="https://support.bioconductor.org/p/90063/">问题</a>，问的是他想要计算人类和老鼠的直系同源基因通过GO注释来计算语义相似性，问GOSemSim是否支持，这个答案是yes and no。</p>

<p></p>

<p>本身是没有直接支持跨物种计算的，GOSemSim最初支持5个物种，后面扩展为20个物种，现在是支持有OrgDb的所有物种，而OrgDb是可以通过AnnotationHub在线获取的，也可以通过AnnotationForge::makeOrgPackage()函数或者是makeOrgPackageFromNCBI()函数构建，所以现在基本上GOSemSim和clusterProfiler都是可以说支持所有有注释的物种，当然没注释的也可以自己注释，通过拿来计算。</p>

<p>GOSemSim默认是支持单一物种，首先通过godata()函数把OrgDb转化为GOSemSim所需要的注释，包括基因映射到GO以及GOID的信息含量，然后这个背景注释传给geneSim/mgeneSim函数，godata()只支持传入单一OrgDb，所以这里只能是支持单一物种，要跨物种也不是不可能，我再写一个merge函数，把godata()输出的GOSemSimDATA对象合并起来，比如一个人的，一个老鼠的，merge完之后，你还可以再merge，这样就可以有多个物种的注释在里面。就可以传给geneSim/mgeneSim来计算了，所有的函数都不需要做出改变，只需要添加这一功能就好。对于使用GO图结构的算法，这没什么问题，但对于使用信息含量的算法，merge的过程，可能需要对信息含量重新计算/处理，这也是我迟迟没写扩展的原因，因为没时间研究这些-,-</p>

<p>虽然直接是不支持的，至少目前是这样（以后可以真会加入像上面提到的功能），但稍微变通一下，就知道其实是可以的，因为你可以使用图结构算法（不涉及信息含量的计算），然后用mgoSim来计算啊，只不过你需要自己做基因到GO的映射而已，你拿一人的基因，对应一个GOID向量，一个老鼠基因，对应另一个GOID向量，做为输入给mgoSim，就可以算出来这两个基因的相似性，简单的wrapper函数，就可以干这个事情了。</p>

 
        </div>
        

<footer>
  <p class="meta">
    <span class="byline author vcard">Posted by <span class="fn">Guangchuang Yu</span></span>
    
    <time>Dec 7, 2016</time>
    <span class="categories">
      Tags:
        
        
          <a class="category" href="https://guangchuangyu.github.io/tags/gosemsim">GOSemSim</a>  
        
    </span>
  </p>

  
  

  

  <p class="meta">
    
        <a class="basic-alignment left" href="https://guangchuangyu.github.io/cn/2016/12/using-linked-list-to-solve-josephus-problem/" title="Using Linked List to solve Josephus problem">Using Linked List to solve Josephus problem</a>
    

    
      <a class="basic-alignment right" href="https://guangchuangyu.github.io/cn/2016/12/clifun%E5%9C%A8r%E9%87%8C%E8%A3%85%E8%A3%85%E9%80%BC/" title="clifun:在R里装装逼">clifun:在R里装装逼</a>
    
  </p>


  

  
</footer>

      </article>
    </div>
    


<aside class="sidebar thirds">
  <section class="first odd">
    

    <p>
      
      
      <script src="//tjs.sjs.sinajs.cn/open/api/js/wb.js" type="text/javascript" charset="utf-8"></script>
      关注Y叔微信公众账号biobabble<br>
      <img src="//guangchuangyu.github.io/blog_images/biobabble.jpg" width="250" height="250" />
      <wb:follow-button uid="2185528715" type="red_3" width="100%" height="24" ></wb:follow-button>

      

    
  </section>



  



    
     
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/05/viziometric/">viziometric</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/05/enrichdavid%E6%94%AF%E6%8C%81%E8%87%AA%E5%AE%9A%E4%B9%89%E8%83%8C%E6%99%AF/">enrichDAVID支持自定义背景</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/05/bootstrap-value%E5%88%86%E6%AE%B5%E6%A0%87%E8%AE%B0/">bootstrap value分段标记</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/04/annotate-tree-with-user-specific-data/">annotate tree with user specific data</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/04/%E5%85%AC%E5%8F%B7%E7%8B%97%E7%9A%84%E7%A6%8F%E9%9F%B3/">公号狗的福音</a>
      </li>
      
    </ul>
  </section>

  
  
  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>

  </section>




  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'GuangchuangYu/Bioinformatics'
  };
</script>
  <script src="//sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
  
</aside>

  </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2017 Guangchuang Yu - <a href="https://guangchuangyu.github.io/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>

