<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width,minimum-scale=1,maximum-scale=1">

  
  <link href="//fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
  <link href="//fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">

  
  <title>Guangchuang Yu</title>

  
  
  <link rel="stylesheet" href="https://guangchuangyu.github.io/cn/css/hugo-octopress.css">

  
  

  
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">

  
  <link href="https://guangchuangyu.github.io/cn/favicon.png" rel="icon">

  
  
  

  <link href="https://guangchuangyu.github.io/cn/index.xml" rel="alternate" type="application/rss+xml" title="Guangchuang Yu" />

  <meta name="description" content="">
  <meta name="keywords" content="">

  <meta name="author" content="Guangchuang Yu">

  
  <meta name="generator" content="Hugo 0.15" />

  
  
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','//www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-77901140-1', 'auto');
ga('send', 'pageview');
</script>


</head>
<body>


<header role="banner"><hgroup>
  
  <h1><a href="https://guangchuangyu.github.io/cn/">Guangchuang Yu</a></h1>
    <h2>a senior-in-age-but-not-senior-in-knowledge bioinformatician</h2>
</hgroup></header>


<nav role="navigation">

<ul class="main-navigation">
  
  
    
      <li><a href="https://guangchuangyu.github.io/cn/" target="_blank" title="Blog">Blog</a></li>
    
  
    
      <li><a href="https://guangchuangyu.github.io/resume" target="_blank" title="Resume">Resume</a></li>
    
  
    
      <li><a href="https://guangchuangyu.github.io/cn/douban" target="_blank" title="Douban">Douban</a></li>
    
  
    
      <li><a href="https://guangchuangyu.github.io/cn/archive" target="_blank" title="Archive">Archive</a></li>
    
  
</ul>


<ul class="subscription">
  <a href="https://guangchuangyu.github.io/cn/index.xml" target="_blank" type="application/rss+xml" title="RSS"><i class="fa fa-rss-square fa-lg"></i></a>

  
  

</ul>


</nav>


<div id="main">
 <div id="content">
  <div class="blog-index">
    
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="https://guangchuangyu.github.io/cn/2017/03/how-to-bug-author/">how to bug author</a>
        </h1>
        <p class="meta">Mar 20, 2017 - 1 minute read -  <a href="https://guangchuangyu.github.io/cn/2017/03/how-to-bug-author/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="https://guangchuangyu.github.io/cn/categories/r/">R</a>
        </p>

      </header>

        
          <p><blockquote>
<p><a href="http://blog.devtang.com/2017/03/05/how-to-get-help/">http://blog.devtang.com/2017/03/05/how-to-get-help/</a> 这篇文章讲问问题的礼仪，会问问题的人容易得到别人的帮助，并不是作者拽，而是对着一群不会问问题，且一副理所当然的用户，早就有了深深的无力感，也看看我写的这篇吧，<a href="https://guangchuangyu.github.io/2016/07/how-to-bug-author/">https://guangchuangyu.github.io/2016/07/how-to-bug-author/</a>.</p>
</blockquote>

<p>这是前几天的文字推送，不知道大家有没有阅读这两篇博客文？如何提问，这是一项重要的技能，很遗憾很多人并没有这项技能！</p>

<p><img src="https://guangchuangyu.github.io/blog_images/2017/413360782.jpg" alt="" /></p>

<p>就在上周，我正好就在微信群里跟某位老师说不要给我个截屏，说能不能在ggtree里实现某功能。懂的人知道，我只是希望大家用正确的方式来做事情，不懂的人，当然觉得我拽，JJYY什么的。</p>
</p>

          
              <footer>
                  <a href="https://guangchuangyu.github.io/cn/2017/03/how-to-bug-author/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="https://guangchuangyu.github.io/cn/2017/03/peak%E6%B3%A8%E9%87%8A/">peak注释</a>
        </h1>
        <p class="meta">Mar 16, 2017 - 1 minute read -  <a href="https://guangchuangyu.github.io/cn/2017/03/peak%E6%B3%A8%E9%87%8A/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="https://guangchuangyu.github.io/cn/categories/r/">R</a><a class="label" href="https://guangchuangyu.github.io/cn/categories/chipseq/">ChIPseq</a>
        </p>

      </header>

        
          <p><p>这一次讲解非常重要的peak注释，注释在<strong>ChIPseeker</strong>里只需要用到一个函数<strong>annotatePeak</strong>，它可以满足大家各方面的需求。</p>

<h2 id="输入:4b21ea5970cfe309bb59877ac4e1f485">输入</h2>

<p>当然需要我们上次讲到的BED文件，<strong>ChIPseeker</strong>自带了5个BED文件，用<strong>getSampleFiles()</strong>可以拿到文件的全路径，它返回的是个named list，我这里取第4个文件来演示。<strong>annotatePeak</strong>的输入也可以是<strong>GRanges</strong>对象，你如果用R做peak calling的话,直接就可以衔接上<strong>ChIPseeker</strong>了。</p>

<pre><code>&gt; require(ChIPseeker)
&gt; f = getSampleFiles()[[4]]
</code></pre>

<p>巧妇难为无米之炊，就像拿到<strong>fastq</strong>要跑<strong>BWA</strong>，你需要全基因组的序列一样，做注释当然需要注释信息，基因的起始终止，基因有那些内含子，外显子，以及它们的起始终止，非编码区的位置，功能元件的位置等各种信息。</p>

<p>很多软件会针对特定的物种去整理这些信息供软件使用，但这样就限制了软件的物种支持，有些开发者写软件本意也是解决自己的问题，可能对自己的研究无关的物种也没兴趣去支持。</p>

<p>然而<strong>ChIPseeker</strong>支持所有的物种，你没有看错，<strong>ChIPseeker</strong>没有物种限制，当然这是有前提的，物种本身起码是有基因的位置这些注释信息，不然就变无米之炊了。</p>

<p>这里我们需要的是一个<strong>TxDb</strong>对象，这个<strong>TxDb</strong>就包含了我们需要的各种信息，<strong>ChIPseeker</strong>会把信息抽取出来，用于注释时使用。</p>

<pre><code>&gt; require(TxDb.Hsapiens.UCSC.hg19.knownGene)
&gt; txdb = TxDb.Hsapiens.UCSC.hg19.knownGene
&gt; x = annotatePeak(f, tssRegion=c(-1000, 1000), TxDb=txdb)
&gt;&gt; loading peak file...				 2017-03-09 11:29:18 PM 
&gt;&gt; preparing features information...		 2017-03-09 11:29:18 PM 
&gt;&gt; identifying nearest features...		 2017-03-09 11:29:19 PM 
&gt;&gt; calculating distance from peak to TSS...	 2017-03-09 11:29:20 PM 
&gt;&gt; assigning genomic annotation...		 2017-03-09 11:29:20 PM 
&gt;&gt; assigning chromosome lengths			 2017-03-09 11:29:42 PM 
&gt;&gt; done...					 2017-03-09 11:29:42 PM 
</code></pre>

<blockquote>
<p>这里需要注意的是，启动子区域是没有明确的定义的，所以你可能需要指定<strong>tssRegion</strong>，把基因起始转录位点的上下游区域来做为启动子区域。</p>
</blockquote>

<p>有了这两个输入（BED文件和TxDb对象），你就可以跑注释了，然后就可以出结果了。</p>
</p>

          
              <footer>
                  <a href="https://guangchuangyu.github.io/cn/2017/03/peak%E6%B3%A8%E9%87%8A/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="https://guangchuangyu.github.io/cn/2017/03/bed%E6%96%87%E4%BB%B6/">BED文件</a>
        </h1>
        <p class="meta">Mar 9, 2017 - 1 minute read -  <a href="https://guangchuangyu.github.io/cn/2017/03/bed%E6%96%87%E4%BB%B6/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="https://guangchuangyu.github.io/cn/categories/r/">R</a>
        </p>

      </header>

        
          <p><p>BED的全称是Browser Extensible Data，顾名思义是为genome browser设计的，大名鼎鼎的bedtools可不是什么「床上用品」。</p>

<p>BED包含有3个必须的字段和9个可选字段。</p>

<p>三个字段包括：</p>

<ul>
<li>1 chrom - 染色体名字</li>
<li>2 chromStart - 染色体起始位点</li>
<li>3 chromEnd - 染色体终止位点</li>
</ul>

<blockquote>
<p>这里必须指出的是<strong>chromStart</strong>是起始于0，而不是1。很多分析软件都忽略
了这一点，会有一个碱基的位移，据我所知Homer和ChIPseeker没有这个问题，而像peakAnalyzer, ChIPpeakAnno等都有位移的问题。</p>
</blockquote>

<p>可选的9个字段包括：</p>

<ul>
<li>4 name - 名字</li>
<li>5 score - 分值(0-1000), 用于genome browser展示时上色。</li>
<li>6 strand - 正负链，对于ChIPseq数据来说，<strong>一般</strong>没有正负链信息。</li>
<li>7 thickStart - 画矩形的起点</li>
<li>8 thickEnd - 画矩形的终点</li>
<li>9 itemRgb - RGB值</li>
<li>10 blockCount - 子元件（比如外显子）的数目</li>
<li>11 blockSizes - 子元件的大小</li>
<li>12 blockStarts - 子元件的起始位点</li>
</ul>
</p>

          
              <footer>
                  <a href="https://guangchuangyu.github.io/cn/2017/03/bed%E6%96%87%E4%BB%B6/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="https://guangchuangyu.github.io/cn/2017/03/linux%E7%89%88%E8%BF%85%E9%9B%B7/">linux版迅雷</a>
        </h1>
        <p class="meta">Mar 6, 2017 - 1 minute read -  <a href="https://guangchuangyu.github.io/cn/2017/03/linux%E7%89%88%E8%BF%85%E9%9B%B7/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="https://guangchuangyu.github.io/cn/categories/sofware/">sofware</a>
        </p>

      </header>

        
          <p><p>上次讲了<a href="http://guangchuangyu.github.io/2016/03/yet-an-unofficial-bioedit-for-osx/">OSX版BioEdit</a>，还蛮受欢迎，说到下载工具，linux的小伙伴们都想用迅雷，有没有？毕竟大家都想耍流氓。许久没打开迅雷，今日一打开，发现被学校给墙掉了。<strong>大写的杯具啊！</strong></p>

<blockquote>
<p>You have attempted to use an application which is in violation of your internet usage policy.</p>

<p>Thunder.Xunlei</p>

<p>Category: P2P</p>
</blockquote>

<p>老司机不能偷偷地下片 了，唯有分享出来，带大家上车。还好我有一张半年前的截屏，不然连张图都没有。</p>
</p>

          
              <footer>
                  <a href="https://guangchuangyu.github.io/cn/2017/03/linux%E7%89%88%E8%BF%85%E9%9B%B7/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    <article>
      <header>
        <h1 class="entry-title">
          <a href="https://guangchuangyu.github.io/cn/2017/03/unknown-annotation-in-enrichment-analysis/">unknown annotation in enrichment analysis</a>
        </h1>
        <p class="meta">Mar 3, 2017 - 1 minute read -  <a href="https://guangchuangyu.github.io/cn/2017/03/unknown-annotation-in-enrichment-analysis/#disqus_thread">Comments</a>

        
        
        
        <a class="label" href="https://guangchuangyu.github.io/cn/categories/statistics/">statistics</a>
        </p>

      </header>

        
          <p><p>小伙伴说注释不全，比如KEGG只有不到1万个基因有注释，但他一次RNA-seq出来的基因有2万多个，那其它没注释那1万多个岂不是扔了？！就某个通路来说，两种情况，要么属于，要么不属于这个通路。那1万多个应该放在背景里，不要扔。</p>

<p>我的解答是三种情况，1属于，2不属于，3不知道。对于缺失信息的，当然是扔。</p>
</p>

          
              <footer>
                  <a href="https://guangchuangyu.github.io/cn/2017/03/unknown-annotation-in-enrichment-analysis/" rel="full-article">Read On &rarr;</a>
              </footer>
          

        

    </article>
    
    
    




<div class="pagination">

    
        
        
        
        <a href="https://guangchuangyu.github.io/cn/" aria-label="First" class="label-pagination"><i class="fa fa-angle-double-left fa-lg"></i></a>
    

    
    
        <a href="https://guangchuangyu.github.io/cn/" aria-label="Previous" class="label-pagination"><i class="fa fa-angle-left fa-lg"></i></a>
    

    
        <a href="https://guangchuangyu.github.io/cn/" class="label-pagination">1</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/2/" class="label-pagination">2</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/3/" class="label-pagination">3</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/4/" class="label-pagination">4</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/5/" class="label-pagination">5</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/6/" class="label-pagination">6</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/7/" class="label-pagination">7</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/8/" class="label-pagination">8</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/9/" class="label-pagination">9</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/10/" class="label-pagination">10</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/11/" class="label-pagination">11</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/12/" class="label-pagination">12</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/13/" class="label-pagination">13</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/14/" class="label-pagination">14</a>
    
        <a href="https://guangchuangyu.github.io/cn/page/15/" class="label-pagination">15</a>
    

    
    
        <a href="https://guangchuangyu.github.io/cn/page/3/" aria-label="Next" class="label-pagination"><i class="fa fa-angle-right fa-lg"></i></a>
    

    
    
        <a href="https://guangchuangyu.github.io/cn/page/15/" aria-label="Last"><i class="fa fa-angle-double-right fa-lg"></i></a>
    

</div>

   

  </div>

  


<aside class="sidebar thirds">
  <section class="first odd">
    

    <p>
      
      
      <script src="//tjs.sjs.sinajs.cn/open/api/js/wb.js" type="text/javascript" charset="utf-8"></script>
      关注Y叔微信公众账号biobabble<br>
      <img src="//guangchuangyu.github.io/blog_images/biobabble.jpg" width="250" height="250" />
      <wb:follow-button uid="2185528715" type="red_3" width="100%" height="24" ></wb:follow-button>

      

    
  </section>



  



    
     
  
  
  <section class="even">
    <h1>Recent Posts</h1>

    
    <ul id="recent_posts">
      
      
        
      
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/04/ggimageggplot2%E4%B8%AD%E6%84%89%E5%BF%AB%E5%9C%B0%E4%BD%BF%E7%94%A8%E5%9B%BE%E7%89%87/">ggimage：ggplot2中愉快地使用图片</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/04/generate-biobabble-banner-using-ggimage/">generate biobabble banner using ggimage</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/04/ggplot2%E5%AD%97%E4%BD%93%E6%BA%A2%E5%87%BA%E7%9A%84%E9%82%A3%E7%82%B9%E7%A0%B4%E4%BA%8B/">ggplot2字体溢出的那点破事</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/03/%E5%85%B3%E4%BA%8Echipseq%E6%B3%A8%E9%87%8A%E7%9A%84%E5%87%A0%E4%B8%AA%E9%97%AE%E9%A2%98/">关于ChIPseq注释的几个问题</a>
      </li>
      
      <li class="post">
        <a href="https://guangchuangyu.github.io/cn/2017/03/news-in-emojifont/">News in emojifont</a>
      </li>
      
    </ul>
  </section>

  
  
  <section class="even">
    <h1>Book to Read</h1>
    Analysis of Phylogenetic and Evolution with R<br>
    Inferring phylogenies<br>
    Statistical Models: Theory and Practice<br>
    Algorithms on Strings, Trees, and Sequences
    <h1>Links</h1>
    <a href="http://www.biostathandbook.com/">Handbook of Biological Statistics</a><br>
    <a href="http://r-bloggers.com/">R-bloggers</a>
    <h1>友邻</h1>
    <a href="http://azaleasays.com/">azalea</a><br>
    <a href="http://blog.shenwei.me/">shenwei</a><br>
    <a href="http://ztpala.com/">ztpala</a>

  </section>




  <script type="text/javascript" src="//rf.revolvermaps.com/0/0/7.js?i=5qpvv5zz7fp&amp;m=0&amp;c=ff0000&amp;cr1=ffffff&amp;sx=0" async="async"></script>

  <script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'GuangchuangYu/Bioinformatics'
  };
</script>
  <script src="//sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
  
</aside>
    
 </div>
</div>




<footer role="contentinfo">
  <p>Copyright &copy; 2017 Guangchuang Yu - <a href="https://guangchuangyu.github.io/cn/license/">License</a> - 
  <span class="credit">Powered by <a target="_blank" href="https://gohugo.io">Hugo</a> and <a target="_blank" href="https://github.com/parsiya/hugo-octopress/">Hugo-Octopress</a> theme.
</p>

</footer>


<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/styles/zenburn.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.1.0/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<script type="text/javascript"
  src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    displayMath: [['$$','$$'], ['\[','\]']],
    processEscapes: true,
    processEnvironments: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
    TeX: { equationNumbers: { autoNumber: "AMS" },
         extensions: ["AMSmath.js", "AMSsymbols.js"] }
  }
});
</script>

<script type="text/javascript" src="//w.sharethis.com/button/buttons.js"></script>
<script type="text/javascript">stLight.options({publisher: "d135f460-3fc5-4802-8169-bd08e4734a09", doNotHash: false, doNotCopy: false, hashAddressBar: false});</script>



</body>
</html>
   