

  
    
  


  




  


  

<!DOCTYPE html>
<html lang="en-us">
  <head>
    
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="Hugo 0.49 with theme Tranquilpeak 0.4.3-BETA">
    <title>非模式基因GO富集分析：以玉米为例&#43;使用OrgDb</title>
    <meta name="author" content="Guangchuang Yu">
    <meta name="keywords" content="">

    <link rel="icon" href="https://guangchuangyu.github.io/cn/favicon.png">
    

    
    <meta name="description" content="模式生物做什么都简单，非模式生物则很多缺少注释，没有注释你就没法做，只能是借助于各种软件比如blastgo，自己跑电子注释。但今天要讲的不是这种情况，很多物种还是有注释的，只是你有时候不知道该去那里下载，或者你有数据，却不知道该怎么用！很多的软件都是针对模式生物的，或者针对某一些类型的非模式生物，能够支持多种非模式生物，能够支持用户自己的注释文件的软件相对来讲，就非常少有了，然而clusterProfiler就是这类少有的软件之一。

获得OrgDb

今天要讲的是通过OrgDb来做GO分析，这是clusterProfiler的enrichGO函数所支持的背景注释，Bioconductor自带20个OrgDb可供使用，多半是模式生物，难道我们要做的物种不在这20个里面就不行了吗？显然不是的，clusterProfiler能支持的物种我自己都数不过来。

我们可以通过AnnotationHub在线检索并抓取OrgDb，比如这里以玉米为例：

">
    <meta property="og:description" content="模式生物做什么都简单，非模式生物则很多缺少注释，没有注释你就没法做，只能是借助于各种软件比如blastgo，自己跑电子注释。但今天要讲的不是这种情况，很多物种还是有注释的，只是你有时候不知道该去那里下载，或者你有数据，却不知道该怎么用！很多的软件都是针对模式生物的，或者针对某一些类型的非模式生物，能够支持多种非模式生物，能够支持用户自己的注释文件的软件相对来讲，就非常少有了，然而clusterProfiler就是这类少有的软件之一。

获得OrgDb

今天要讲的是通过OrgDb来做GO分析，这是clusterProfiler的enrichGO函数所支持的背景注释，Bioconductor自带20个OrgDb可供使用，多半是模式生物，难道我们要做的物种不在这20个里面就不行了吗？显然不是的，clusterProfiler能支持的物种我自己都数不过来。

我们可以通过AnnotationHub在线检索并抓取OrgDb，比如这里以玉米为例：

">
    <meta property="og:type" content="blog">
    <meta property="og:title" content="非模式基因GO富集分析：以玉米为例&#43;使用OrgDb">
    <meta property="og:url" content="/2017/07/clusterprofiler-maize/">
    <meta property="og:site_name" content="Guangchuang Yu">
    <meta name="twitter:card" content="summary">
    <meta name="twitter:title" content="Guangchuang Yu">
    <meta name="twitter:description" content="模式生物做什么都简单，非模式生物则很多缺少注释，没有注释你就没法做，只能是借助于各种软件比如blastgo，自己跑电子注释。但今天要讲的不是这种情况，很多物种还是有注释的，只是你有时候不知道该去那里下载，或者你有数据，却不知道该怎么用！很多的软件都是针对模式生物的，或者针对某一些类型的非模式生物，能够支持多种非模式生物，能够支持用户自己的注释文件的软件相对来讲，就非常少有了，然而clusterProfiler就是这类少有的软件之一。

获得OrgDb

今天要讲的是通过OrgDb来做GO分析，这是clusterProfiler的enrichGO函数所支持的背景注释，Bioconductor自带20个OrgDb可供使用，多半是模式生物，难道我们要做的物种不在这20个里面就不行了吗？显然不是的，clusterProfiler能支持的物种我自己都数不过来。

我们可以通过AnnotationHub在线检索并抓取OrgDb，比如这里以玉米为例：

">
    
      <meta name="twitter:creator" content="@guangchuangyu">
    
    

    
    

    
      <meta property="og:image" content="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=640">
    

    
    
    

    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/jquery.fancybox.min.css" integrity="sha256-vuXZ9LGmmwtjqFX1F+EKin1ThZMub58gKULUyf0qECk=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.4/helpers/jquery.fancybox-thumbs.min.css" integrity="sha256-SEa4XYAHihTcEP1f5gARTB2K26Uk8PsndQYHQC1f4jU=" crossorigin="anonymous" />
    
    
    <link rel="stylesheet" href="https://guangchuangyu.github.io/cn/css/style-jsjn0006wyhpyzivf6yceb31gvpjatbcs3qzjvlumobfnugccvobqwxnnaj8.min.css" />
    
    

    
      
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-77901140-1', 'auto');
	
	ga('send', 'pageview');
}
</script>

    
    
  </head>

  <body>
    <div id="blog">
      <header id="header" data-behavior="4">
  <i id="btn-open-sidebar" class="fa fa-lg fa-bars"></i>
  <div class="header-title">
    <a class="header-title-link" href="https://guangchuangyu.github.io/cn/">Guangchuang Yu</a>
  </div>
  
    
      <a class="header-right-picture "
         href="https://guangchuangyu.github.io/cn/#about">
    
    
    
      
        <img class="header-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=90" alt="Author&#39;s picture" />
      
    
    </a>
  
</header>

      <nav id="sidebar" data-behavior="4">
  <div class="sidebar-container">
    
      <div class="sidebar-profile">
        <a href="https://guangchuangyu.github.io/cn/#about">
          <img class="sidebar-profile-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
        </a>
        <h4 class="sidebar-profile-name">Guangchuang Yu</h4>
        
          <h5 class="sidebar-profile-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</h5>
        
      </div>
    
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/categories">
    
      <i class="sidebar-button-icon fa fa-lg fa-bookmark"></i>
      
      <span class="sidebar-button-desc">Categories</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/tags">
    
      <i class="sidebar-button-icon fa fa-lg fa-tags"></i>
      
      <span class="sidebar-button-desc">Tags</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/archives">
    
      <i class="sidebar-button-icon fa fa-lg fa-archive"></i>
      
      <span class="sidebar-button-desc">Archives</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/#about">
    
      <i class="sidebar-button-icon fa fa-lg fa-question"></i>
      
      <span class="sidebar-button-desc">About</span>
    </a>
  </li>

  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/">
    
      <i class="sidebar-button-icon fa fa-lg fa-home"></i>
      
      <span class="sidebar-button-desc">Home</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://github.com/GuangchuangYu" target="_blank" rel="noopener">
    
      <i class="sidebar-button-icon fa fa-lg fa-github"></i>
      
      <span class="sidebar-button-desc">GitHub</span>
    </a>
  </li>


    </ul>
    <ul class="sidebar-buttons">
      
  <li class="sidebar-button">
    
      <a class="sidebar-button-link " href="https://guangchuangyu.github.io/cn/index.xml">
    
      <i class="sidebar-button-icon fa fa-lg fa-rss"></i>
      
      <span class="sidebar-button-desc">RSS</span>
    </a>
  </li>


    </ul>
  </div>
</nav>

      

      <div id="main" data-behavior="4"
        class="
               hasCoverMetaIn
               ">
        <article class="post" itemscope itemType="http://schema.org/BlogPosting">
          
          
            <div class="post-header main-content-wrap text-left">
  
    <h1 class="post-title" itemprop="headline">
      非模式基因GO富集分析：以玉米为例&#43;使用OrgDb
    </h1>
  
  
  <div class="postShorten-meta post-meta">
    
      <time itemprop="datePublished" datetime="2017-07-24T08:53:00&#43;08:00">
        
  July 24, 2017

      </time>
    
    
  
  
    <span>in</span>
    
      <a class="category-link" href="https://guangchuangyu.github.io/cn/categories/r">R</a>
    
  

  </div>

</div>
          
          <div class="post-content markdown" itemprop="articleBody">
            <div class="main-content-wrap">
              <p>模式生物做什么都简单，非模式生物则很多缺少注释，没有注释你就没法做，只能是借助于各种软件比如blastgo，自己跑电子注释。但今天要讲的不是这种情况，很多物种还是有注释的，只是你有时候不知道该去那里下载，或者你有数据，却不知道该怎么用！很多的软件都是针对模式生物的，或者针对某一些类型的非模式生物，能够支持多种非模式生物，能够支持用户自己的注释文件的软件相对来讲，就非常少有了，然而clusterProfiler就是这类少有的软件之一。</p>

<h2 id="获得orgdb">获得OrgDb</h2>

<p>今天要讲的是通过OrgDb来做GO分析，这是clusterProfiler的<strong>enrichGO</strong>函数所支持的背景注释，Bioconductor自带20个OrgDb可供使用，多半是模式生物，难道我们要做的物种不在这20个里面就不行了吗？显然不是的，clusterProfiler能支持的物种我自己都数不过来。</p>

<p>我们可以通过AnnotationHub在线检索并抓取OrgDb，比如这里以玉米为例：</p>

<p></p>

<pre><code>&gt; require(AnnotationHub)
&gt; hub &lt;- AnnotationHub()
&gt; query(hub, &quot;zea&quot;)
AnnotationHub with 2 records
# snapshotDate(): 2017-04-25 
# $dataprovider: Inparanoid8, ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/
# $species: Gibberella zeae, Zea mays
# $rdataclass: Inparanoid8Db, OrgDb
# additional mcols(): taxonomyid, genome, description,
#   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
#   rdatapath, sourceurl, sourcetype 
# retrieve records with, e.g., 'object[[&quot;AH10514&quot;]]' 

            title                          
  AH10514 | hom.Gibberella_zeae.inp8.sqlite
  AH55736 | org.Zea_mays.eg.sqlite   
</code></pre>

<p>通过检索，<code>org.Zea_mays.eg.sqlite</code>就是我们所要的OrgDb，可以通过相应的accession number, <strong>AH55736</strong>抓取文件，并存入了maize对象中，它包含了51097个基因的注释：</p>

<pre><code>&gt; maize &lt;- hub[['AH55736']]
&gt; length(keys(maize))
[1] 51097
</code></pre>

<p>这个OrgDb，包含有以下一些注释信息：</p>

<pre><code>&gt; columns(maize)
 [1] &quot;ACCNUM&quot;      &quot;ALIAS&quot;       &quot;CHR&quot;         &quot;ENTREZID&quot;    &quot;EVIDENCE&quot;   
 [6] &quot;EVIDENCEALL&quot; &quot;GENENAME&quot;    &quot;GID&quot;         &quot;GO&quot;          &quot;GOALL&quot;      
[11] &quot;ONTOLOGY&quot;    &quot;ONTOLOGYALL&quot; &quot;PMID&quot;        &quot;REFSEQ&quot;      &quot;SYMBOL&quot;     
[16] &quot;UNIGENE&quot;
</code></pre>

<h2 id="转换id">转换ID</h2>

<p>我们可以使用bitr来转换ID，甚至于直接检索GO注释：</p>

<pre><code>&gt; require(clusterProfiler)
&gt; bitr(keys(maize)[1], 'ENTREZID', c(&quot;REFSEQ&quot;, &quot;GO&quot;, &quot;ONTOLOGY&quot;), maize)
   ENTREZID         REFSEQ         GO ONTOLOGY
1    541612 XP_008648268.1 GO:0009507       CC
2    541612 XP_008648268.1 GO:0051537       MF
3    541612 XP_008648268.1 GO:0009055       MF
4    541612 XP_008648268.1 GO:0046872       MF
5    541612 XP_008648268.1 GO:0022900       BP
6    541612 NP_001104837.2 GO:0009507       CC
7    541612 NP_001104837.2 GO:0051537       MF
8    541612 NP_001104837.2 GO:0009055       MF
9    541612 NP_001104837.2 GO:0046872       MF
10   541612 NP_001104837.2 GO:0022900       BP
11   541612 XM_008650046.2 GO:0009507       CC
12   541612 XM_008650046.2 GO:0051537       MF
13   541612 XM_008650046.2 GO:0009055       MF
14   541612 XM_008650046.2 GO:0046872       MF
15   541612 XM_008650046.2 GO:0022900       BP
16   541612 NM_001111367.2 GO:0009507       CC
17   541612 NM_001111367.2 GO:0051537       MF
18   541612 NM_001111367.2 GO:0009055       MF
19   541612 NM_001111367.2 GO:0046872       MF
20   541612 NM_001111367.2 GO:0022900       BP
</code></pre>

<h2 id="go富集分析">GO富集分析</h2>

<pre><code>&gt; sample_genes &lt;- keys(maize)[1:100]
&gt; head(sample_genes)
[1] &quot;541612&quot; &quot;541613&quot; &quot;541614&quot; &quot;541615&quot; &quot;541617&quot; &quot;541618&quot;
</code></pre>

<p>这里我只是简单地使用ID列表中前100个ENTREZ基因ID，也可以使用其它的ID，通过借助于<strong>bitr</strong>进行转换，或者通过给<strong>enrichGO</strong>指定ID类型(keyType参数）。</p>

<p>有了OrgDb，使用起来，就跟文档中使用人类基因做为例子一样，用法一致，并且也可以通过clusterProfiler所提供的各种可视化函数对结果进行展示：</p>

<pre><code>&gt; require(clusterProfiler)
&gt; res = enrichGO(sample_genes, OrgDb=maize, pvalueCutoff=1, qvalueCutoff=1)
&gt; res
#
# over-representation test
#
#...@organism 	 Zea mays 
#...@ontology 	 MF 
#...@keytype 	 ENTREZID 
#...@gene 	 chr [1:100] &quot;541612&quot; &quot;541613&quot; &quot;541614&quot; &quot;541615&quot; &quot;541617&quot; &quot;541618&quot; &quot;541619&quot; ...
#...pvalues adjusted by 'BH' with cutoff &lt;1 
#...114 enriched terms found
'data.frame':	114 obs. of  9 variables:
 $ ID         : chr  &quot;GO:0004871&quot; &quot;GO:0000155&quot; &quot;GO:0004673&quot; &quot;GO:0016775&quot; ...
 $ Description: chr  &quot;signal transducer activity&quot; &quot;phosphorelay sensor kinase activity&quot; &quot;protein histidine kinase activity&quot; &quot;phosphotransferase activity, nitrogenous group as acceptor&quot; ...
 $ GeneRatio  : chr  &quot;9/80&quot; &quot;5/80&quot; &quot;5/80&quot; &quot;5/80&quot; ...
 $ BgRatio    : chr  &quot;81/14230&quot; &quot;22/14230&quot; &quot;23/14230&quot; &quot;23/14230&quot; ...
 $ pvalue     : num  6.65e-10 1.21e-07 1.54e-07 1.54e-07 1.90e-07 ...
 $ p.adjust   : num  7.58e-08 3.60e-06 3.60e-06 3.60e-06 3.60e-06 ...
 $ qvalue     : num  6.37e-08 3.03e-06 3.03e-06 3.03e-06 3.03e-06 ...
 $ geneID     : chr  &quot;541618/541625/541627/541634/541636/541638/541641/541642/541663&quot; &quot;541627/541634/541641/541642/541663&quot; &quot;541627/541634/541641/541642/541663&quot; &quot;541627/541634/541641/541642/541663&quot; ...
 $ Count      : int  9 5 5 5 4 4 4 5 5 5 ...
#...Citation
  Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.
  clusterProfiler: an R package for comparing biological themes among
  gene clusters. OMICS: A Journal of Integrative Biology
  2012, 16(5):284-287 

</code></pre>

<p>如果你有表达谱数据，你也可以使用gseGO进行GSEA分析，这里我懒得上网找数据来演示了，用法反正跟文档里的一样，只不过换成了你自己的数据和相应物种的OrgDb对象而已。</p>

<h2 id="如果没有orgdb怎么办">如果没有OrgDb怎么办？</h2>

<p>必须也是可以做的，这个将在以后讲解！</p>

<h2 id="相关阅读">相关阅读</h2>

<ul>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzI5NjUyNzkxMg==&amp;mid=2247484398&amp;idx=1&amp;sn=718137b690949ea8e17ff245afa022a5&amp;chksm=ec43b2a9db343bbfc011f1832556320575733a858f4841f1ab8dec4868ca46f85cca0434f559#rd">富集分析</a></li>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzI5NjUyNzkxMg==&amp;mid=2247484374&amp;idx=1&amp;sn=f946f3539a752b09a94b5e160e87368e&amp;chksm=ec43b291db343b871a1f10ca31d6ef94e9da4485043ed374c9739689d02072be92a34ca4e863#rd">clusterProfiler for enrichment analysis</a></li>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzI5NjUyNzkxMg==&amp;mid=2247484414&amp;idx=1&amp;sn=096f758f04f875e25c333c345f530742&amp;chksm=ec43b2b9db343baf4e2a8640f48e7e72b3ef1b057c91c68905debb2bb54e5d2fdda62438628a#rd">why clusterProfiler fails</a></li>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzI5NjUyNzkxMg==&amp;mid=2247484442&amp;idx=1&amp;sn=016b12d91af43e16ac8731ccd154b4b2&amp;chksm=ec43b55ddb343c4be1fb6f5f764a421889b8ba93e5784f4921a6f510ee3572c72d14c76059ae#rd">Comparison of clusterProfiler and GSEA-P</a></li>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzI5NjUyNzkxMg==&amp;mid=2247484096&amp;idx=1&amp;sn=dbee71d4af2578746a1caa8fb0ad55a1&amp;chksm=ec43b387db343a91d48c9f69e85ccb5ba18d44ec025356e9a63705f38e1690c73023b80fdbd1#rd">buildGOmap</a></li>
</ul>
              
            </div>
          </div>
          <div id="post-footer" class="post-footer main-content-wrap">
            
              
                
                
                  <div class="post-footer-tags">
                    <span class="text-color-light text-small">TAGGED IN</span><br/>
                    
  <a class="tag tag--primary tag--small" href="https://guangchuangyu.github.io/cn/tags/clusterprofiler/">clusterProfiler</a>

  <a class="tag tag--primary tag--small" href="https://guangchuangyu.github.io/cn/tags/orgdb/">OrgDb</a>

  <a class="tag tag--primary tag--small" href="https://guangchuangyu.github.io/cn/tags/maize/">maize</a>

                  </div>
                
              
            
            
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
          
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2017/07/scale-size/" data-tooltip="搞大你的点，让我们画真正的气泡图">
              
                  <i class="fa fa-angle-left"></i>
                  <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
                </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2017/07/swirl/" data-tooltip="听说你想学R">
              
                  <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                  <i class="fa fa-angle-right"></i>
                </a>
            </li>
          
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


            
              
                <div id="disqus_thread">
  <noscript>Please enable JavaScript to view the <a href="//disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
              
            
          </div>
        </article>
        <footer id="footer" class="main-content-wrap">
  <span class="copyrights">
    &copy; 2018 Guangchuang Yu. All Rights Reserved
  </span>
</footer>

      </div>
      <div id="bottom-bar" class="post-bottom-bar" data-behavior="4">
        
  <div class="post-actions-wrap">
      <nav >
        <ul class="post-actions post-action-nav">
          
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2017/07/scale-size/" data-tooltip="搞大你的点，让我们画真正的气泡图">
              
                  <i class="fa fa-angle-left"></i>
                  <span class="hide-xs hide-sm text-small icon-ml">NEXT</span>
                </a>
            </li>
            <li class="post-action">
              
                <a class="post-action-btn btn btn--default tooltip--top" href="https://guangchuangyu.github.io/cn/2017/07/swirl/" data-tooltip="听说你想学R">
              
                  <span class="hide-xs hide-sm text-small icon-mr">PREVIOUS</span>
                  <i class="fa fa-angle-right"></i>
                </a>
            </li>
          
        </ul>
      </nav>
    <ul class="post-actions post-action-share" >
      
        <li class="post-action hide-lg hide-md hide-sm">
          <a class="post-action-btn btn btn--default btn-open-shareoptions" href="#btn-open-shareoptions">
            <i class="fa fa-share-alt"></i>
          </a>
        </li>
        
      
      
        <li class="post-action">
          <a class="post-action-btn btn btn--default" href="#disqus_thread">
            <i class="fa fa-comment-o"></i>
          </a>
        </li>
      
      <li class="post-action">
        
          <a class="post-action-btn btn btn--default" href="#">
        
          <i class="fa fa-list"></i>
        </a>
      </li>
    </ul>
  </div>


      </div>
      <div id="share-options-bar" class="share-options-bar" data-behavior="4">
  <i id="btn-close-shareoptions" class="fa fa-close"></i>
  <ul class="share-options">
    
  </ul>
</div>
<div id="share-options-mask" class="share-options-mask"></div>
    </div>
    
    <div id="about">
  <div id="about-card">
    <div id="about-btn-close">
      <i class="fa fa-remove"></i>
    </div>
    
      <img id="about-card-picture" src="//www.gravatar.com/avatar/7466436733e8691a63a9f7af4651e014?s=110" alt="Author&#39;s picture" />
    
    <h4 id="about-card-name">Guangchuang Yu</h4>
    
      <div id="about-card-bio">a senior-in-age-but-not-senior-in-knowledge bioinformatician</div>
    
    
      <div id="about-card-job">
        <i class="fa fa-briefcase"></i>
        <br/>
        Postdoc researcher
      </div>
    
    
      <div id="about-card-location">
        <i class="fa fa-map-marker"></i>
        <br/>
        Hong Kong
      </div>
    
  </div>
</div>

    <div id="algolia-search-modal" class="modal-container">
  <div class="modal">
    <div class="modal-header">
      <span class="close-button"><i class="fa fa-close"></i></span>
      <a href="https://algolia.com" target="_blank" rel="noopener" class="searchby-algolia text-color-light link-unstyled">
        <span class="searchby-algolia-text text-color-light text-small">by</span>
        <img class="searchby-algolia-logo" src="https://www.algolia.com/static_assets/images/press/downloads/algolia-light.svg">
      </a>
      <i class="search-icon fa fa-search"></i>
      <form id="algolia-search-form">
        <input type="text" id="algolia-search-input" name="search"
          class="form-control input--large search-input" placeholder="Search" />
      </form>
    </div>
    <div class="modal-body">
      <div class="no-result text-color-light text-center">no post found</div>
      <div class="results">
        
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/09/proxychains/">
                <h3 class="media-heading">一次解决所有代理问题</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><blockquote>
<p>之前公众号上有人说，我回大陆了会有一波科学上网的教程，今天的推文算一个吧。真的是被你们言中了！</p>
</blockquote>

<p>做为Bioconductor包作者，而<code>git push</code>传代码到Bioconductor上是被封的。这很是恼火，随着10月份将迎来新一版本的Bioconductor发布，不能够push代码是多么惨的一件事。</p>

<p>代理其实也挺恼火的，因为各种协议啊，你要各种设置啊，<code>http/https/ssh/git</code>都是各种配置，其实我只要有一个<code>socks5</code>的代理，然后你们所有需要走代理的程序都通过它就好了。</p>

<p>谁特么有空天天在琢磨怎么配置各种不同协议的代理！只要我们想得到的，一般都有人做了。</p>

<p>于是万能的github上找到了<a href="https://github.com/haad/proxychains">proxychains</a>。</p>

<pre><code>A hook preloader that allows to redirect TCP traffic of existing
dynamically linked programs through one or more SOCKS or HTTP
proxies
</code></pre>

<p>它强制给定程序发起的TCP连接通过事先配置的代理。可以涵盖所有需要代理的情景。就以git为例，没有proxychains的话，就必须为每个协议（<code>https</code>, <code>git</code>, <code>ssh</code>）按照git文档的要求分别设置代理，过程复杂且不稳定。有了<code>proxychains</code>，这些完全可以不用管，当然应用场景不限于<code>git</code>，任何一个需要代理的命令行程序都是一样在运行的指令前面加<code>proxychains</code>完事。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/09/install-bioconductor-package/">
                <h3 class="media-heading">不用biocLite安装Bioconductor包</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>Bioconductor包的标准安装方法是：</p>

<pre><code>source(&quot;https://bioconductor.org/biocLite.R&quot;)
biocLite(pkg_name)
</code></pre>

<p>这其实是非常不好的，多年以来，一直没有改进，试想一下，bioconductor的网站如果被人黑了，放一段危险的代码，然后你却傻傻地去<code>source</code>它，真的是画面太美！</p>

<p>Bioconductor终于是想要改变了，往CRAN上扔了一个叫<code>BiocManager</code>的包，以后通过CRAN安装这个包，再通过这个包来安装Bioconductor的包，显得安全多了，也不用大家费辣么多口水说你得先<code>source</code>一段网上的代码，当然这个<code>BiocManager</code>还不成熟，起码还没有在<code>BiocInstaller</code>加载的时候提示大家转而用<code>BiocManager</code>，也没有在Bioconductor的网页上介绍使用它来安装。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/09/facet_labeller/">
                <h3 class="media-heading">用facet_labeller改facet_plot的panel名</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Sep 9, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><blockquote>
<p>I post this because I am facing the same problem: I want to change the text of the label (not the font, the face, the colour, etc: the text itself) of the leftmost panel which is automatically called &ldquo;Tree&rdquo; by ggtree. As far as I understand, and contrary to what is posted below, the &ldquo;theme&rdquo; thing of ggplot2 only allows to change the appearance of the panel header, not the text of the panel title itself&hellip; So how to go about that?</p>
</blockquote>

<p>这是<a href="https://groups.google.com/d/msgid/bioc-ggtree/154a6acc-bf78-40d9-ad95-8dc1877a4f97%40googlegroups.com?utm_medium=email&amp;utm_source=footer">google group</a>里的问题，说的是<code>facet_plot</code>会把画树的panel叫做Tree，而他想改名，于是我就写了一个函数，<code>facet_labeller</code>来回应这个问题。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/08/ggplot-3d/">
                <h3 class="media-heading">用ggplot2画3D</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Aug 8, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>在《<a href="http://mp.weixin.qq.com/s/BPY4dPFrrNrDKccMvax07w">扪心自问，meme几何？</a>》一文中，我画了一只兔子，今天给大家演示一下画一只3D版本的兔子，我们知道<code>ggplot2</code>不支持3D，只能画2D的图，但3D其实只是视觉上的，显示在你屏幕上的，同样只是2D。所以我们要做的，就是把3D的坐标转换成2D，然后画出2D的图片来，而这2D看上去有3D的效果。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/08/arch-downgrade/">
                <h3 class="media-heading">Arch之软件降级</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Aug 8, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>用Arch Linux就是每天睡觉关机前滚一下，然而有时候有些软件升级滚出问题，比如<code>glibc-2.28</code>就导致了<code>R</code>安装包有问题，会抛出下面的错误：</p>

<pre><code>terminate called after throwing an instance of 'std::runtime_error'
  what():  Mutex creation failed
</code></pre>

<p>用<code>blogdown</code>编译博客也同样会抛出这个错误。</p>

<p>然后<code>electron</code>也有问题，《<a href="http://mp.weixin.qq.com/s/L556YNtVSKf84cEFdEZVBQ">开源的macOS和linux版微信</a>》跑着跑着突然CPU就狂飙。</p>

<p>解决办法就是安装回老版本，但<code>pacman</code>永远是最新的，当然可以去git/svn仓库里找老版本的<code>PKGBUILD</code>，然后再<code>makepkg</code>，但经常会在<code>signature checking</code>这步报错，你还得到导入<code>keyring</code>，挺麻烦。</p>

<p>这时候我找到了<code>downgrade</code>脚本:</p>

<blockquote>
<p>Bash script for downgrading one or more packages to a version in your cache or the A.L.A.</p>
</blockquote>

<p>AUR有收录，安装非常简单：</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/08/pheatmap-class/">
                <h3 class="media-heading">pheatmap类属性：一个更好用的pheatmap</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Aug 8, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p><code>pheatmap</code>我其实用的次数不多，但在帮别人解决一些问题之后，比如：</p>

<ul>
<li><a href="http://mp.weixin.qq.com/s/Xhi_ZkrEzUVdnCu7ECBS4w">听说你还不会画热图</a></li>
<li><a href="https://mp.weixin.qq.com/s/yVLgGQQi0CQ9Q9UfIKl5aA">不同数据集画出的热图，用同样的颜色区间上色?</a></li>
</ul>

<p>我发现这包的交互能力是比较差的，它就像传统的<code>plot</code>函数一样，命令一打下去，就直接画图，虽然它其实是有输出的，但事实上超出了一般用户的理解能力，它的输出，比如说可以用<code>cowplot</code>来拼图，我在上面链接的那篇文章中皆来演示，然而很多人问我，怎么把<code>pheatmap</code>画的热图和<code>ggplot2</code>的图，用<code>cowplot</code>来拼。这就是因为拼图的方法超出了很多的理解范围，看过演示，也只能照猫画老虎，过后也就忘记了。有没有办法用大家可以理解的方法来拼？这样就不太容易忘记，答案是肯定的，我下次再讲。今天先讲一讲其前提条件。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/linux_qq/">
                <h3 class="media-heading">酗酒的Linux不是好企鹅</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>国内的办公环境，离不开QQ等Windows软件，以前我通过《<a href="http://mp.weixin.qq.com/s/7AdkrX3hZYdKNP4YIhgZLA">安装黑苹果</a>》来解决，如果要用Linux呢？我们不禁想问，<a href="https://solarianprogrammer.com/2017/04/16/windows-susbsystem-for-linux-xfce-4/">连Windows都能跑Linux程序了</a>，Linux到底什么时候能跑Windows程序？</p>

<h2 id="干了这杯酒-不干没鸡鸡">干了这杯酒，不干没鸡鸡！</h2>

<p><img src="https://lh3.googleusercontent.com/-dfDJJYSQ4MM/V-PSb7LmfPI/AAAAAAAAEy8/EXNlBjf7GZI_aAfC-lwQx2bHkppC5ereQCJoC/w530-h530-n-rw/e446d718-5ca4-42ec-8d59-45f4630b211d" alt="" /></p>

<p>Wine虽然是一个解决方案，而且现在也有现成的方案，比如《<a href="http://mp.weixin.qq.com/s/4yWzHxjB9WL33ujagi5BAw">你们想要的：QQ跑在Linux上</a> 》，再比如有打包好的深度QQ等。但我不想在我64位的系统里装一堆32位的兼容包，也不想（多多少少得）折腾Wine。</p>

<h2 id="无需安装-开箱即用">无需安装，开箱即用？</h2>

<p>这显然是最好的方案，比如在苹果系统上，我已经打包了《<a href="http://mp.weixin.qq.com/s/F0TxMk1CrYVNvxWqZgv-Aw">OSX版BioEdit</a>》和《<a href="https://mp.weixin.qq.com/s/P3WhPkTrf2dUkzvOba3hlQ">独家：SciHub苹果桌面版</a>》，你只需要拖到<code>Applications</code>里，双击就能用。</p>

<p>这有什么好处？第一你不用去系统里装Wine以及一堆依赖包，第二你不需要去配置Wine，第三这个运行环境是与你的系统分离的（系统是干净的）。对于用户来说，傻瓜化开箱即用，对于系统来说，干净不影响。这是最好的方案。谁愿意去装一堆没用的兼容<code>libs</code>，谁愿意去折腾老半天还用不上！当然有一点点不好的地方，是打包的软件体积稍大一点，因为你要打包整个运行环境嘛，但现在最不值钱的就是存储空间了。</p>

<p>所以最好的方案是像Mac这样，可以把运行环境分离出来，系统干净不折腾！当然再者还得有像我这样的好心人去打包。</p>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/partial-match-when-subsetting-rowname-of-data-frame/">
                <h3 class="media-heading">R总给我惊喜</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><blockquote>
<p>Hello!
First of all, I would like to thank you for this wonderful and very powerful package!</p>

<p>I have tried to plot a phylogenetic tree with heatmap of associated matrix (with <code>gheatmap</code>).
I found that the row names of matrix doesn&rsquo;t exactly match to the tip names of tree in case if there is missing data in associated matrix.
In other words, if we have two species (e.g., Species<strong>198</strong> and Species<strong>1981</strong>), but only one of them is represented in the associated matrix, we will have colored cells for both species in the heatmap.</p>
</blockquote>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/clusterprofiler-ath/">
                <h3 class="media-heading">拟南芥的富集分析</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>「知识星球」里的问题，<a href="https://t.zsxq.com/aEyneA2">https://t.zsxq.com/aEyneA2</a>，拟南芥的ID怎么搞，这个内置是支持的。</p>

<p>比如我随便搞几个ID来演示一下：</p>

<pre><code>&gt; require(clusterProfiler)
Loading required package: clusterProfiler

clusterProfiler v3.9.0  For help: https://guangchuangyu.github.io/software/clusterProfiler

If you use clusterProfiler in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for 
comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.

g &lt;- c(&quot;AT5G61600&quot;, &quot;AT2G44230&quot;, &quot;AT1G49130&quot;, &quot;AT1G30310&quot;, &quot;AT3G18560&quot;,
&quot;AT2G15960&quot;, &quot;AT2G29290&quot;, &quot;AT1G08930&quot;, &quot;AT2G44460&quot;, &quot;AT3G46600&quot;,
&quot;AT5G24380&quot;, &quot;AT3G59270&quot;, &quot;AT5G55780&quot;, &quot;AT2G45090&quot;, &quot;AT5G38430&quot;,
&quot;AT2G17040&quot;, &quot;AT4G35770&quot;, &quot;AT5G30341&quot;, &quot;AT1G56320&quot;, &quot;AT2G16367&quot;)
</code></pre>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
          <div class="media">
            
            <div class="media-body">
              <a class="link-unstyled" href="https://guangchuangyu.github.io/cn/2018/07/scholar-impact-factor/">
                <h3 class="media-heading">scholar包检索期刊影响因子（2018年最新版本）</h3>
              </a>
              <span class="media-meta">
                <span class="media-date text-small">
                  Jul 7, 2018
                </span>
              </span>
              <div class="media-content hide-xs font-merryweather"><p>在《<a href="https://mp.weixin.qq.com/s/kBnLyNl2anXuMa6sDepRQw">一个要复活的R包和一个404的网站</a>》一文中第一次介绍了scholar包，它可以应用于《<a href="https://mp.weixin.qq.com/s/IBrEOh92r_90DRbGb3XlKg">自动抓取论文的引用数据</a>》，虽然数据源来自于Google Scholar，国内用户可能访问不了，但今天要介绍的这个功能，所有人都是可以用的，它允许我们检索期刊的影响因子，在《<a href="https://mp.weixin.qq.com/s/qnuiepL4-nSiSWJVJH54ug">scholar包支持检索期刊影响因子</a>》一文中第一次介绍了这个功能，当时是2016年的影响因子（2017年发布），现在数据已经更新至2018年6月26号发布的2017年影响因子。</p>

<pre><code>&gt; library(scholar)
&gt; jn = c(&quot;bioinformatics&quot;, &quot;methods in ecology and evolution&quot;,
+     &quot;molecular biosystems&quot;, &quot;molecular biology and evolution&quot;)
&gt; get_impactfactor(jn)
# A tibble: 4 x 4
  Journal                          Cites ImpactFactor Eigenfactor
  &lt;chr&gt;                            &lt;dbl&gt;        &lt;dbl&gt;       &lt;dbl&gt;
1 BIOINFORMATICS                   95300         5.48      0.201
2 Methods in Ecology and Evolution  8866         6.36      0.0370
3 Molecular BioSystems              7269         2.76      0.0175
4 MOLECULAR BIOLOGY AND EVOLUTION  44664        10.2       0.102
</code></pre>

<p></p></div>
            </div>
            <div style="clear:both;"></div>
            <hr>
          </div>
        
      </div>
    </div>
    <div class="modal-footer">
      <p class="results-count text-medium"
         data-message-zero="no post found"
         data-message-one="1 post found"
         data-message-other="{n} posts found">
         233 posts found
      </p>
    </div>
  </div>
</div>
    
  
    
    <div id="cover" style="background-image:url('https://guangchuangyu.github.io/cn/images/cover.jpg');"></div>
  


    
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.4/jquery.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/2.1.7/js/jquery.fancybox.min.js" integrity="sha256-GEAnjcTqVP+vBp3SSc8bEDQqvWAZMiHyUSIorrWwH50=" crossorigin="anonymous"></script>


<script src="https://guangchuangyu.github.io/cn/js/script-qi9wbxp2ya2j6p7wx1i6tgavftewndznf4v0hy2gvivk1rxgc3lm7njqb6bz.min.js"></script>




  
    
      <script>
        var disqus_config = function () {
          this.page.url = 'https:\/\/guangchuangyu.github.io\/cn\/2017\/07\/clusterprofiler-maize\/';
          
            this.page.identifier = '\/2017\/07\/clusterprofiler-maize\/'
          
        };
        (function() {
          
          
          if (window.location.hostname == "localhost") {
            return;
          }
          var d = document, s = d.createElement('script');
          var disqus_shortname = 'gcyu';
          s.src = '//' + disqus_shortname + '.disqus.com/embed.js';

          s.setAttribute('data-timestamp', +new Date());
          (d.head || d.body).appendChild(s);
        })();
      </script>
    
  



    
  </body>
</html>

